<!DOCTYPE html><html><head>	<meta charset="utf-8">	<title>generación de archivo html</title></head><body>Small heterodimer partner (SHP; NR0B2) is an unusual orphan nuclear receptor that lacks a conventional DNA-binding domain and acts as a modulator of transcriptional activities of a number of nuclear receptors. Herein, we report that the human SHP promoter (hSHP) is activated by sterol regulatory element-binding protein-1 (SREBP-1), which regulates the expression of various genes involved in cholesterol and fatty acid synthesis. Overexpression of SREBP-1 activated the human but not mouse SHP promoter, although SREBP-2 had little effect on the SHP promoter in CV-1 cells. Serial deletion reporter assays revealed that SREBP-1-responsive region is located within the sequences from -243 to -120 bp in the hSHP promoter. DNase I footprinting, gel shift assays, and chromatin immunoprecipitation assays demonstrated that SREBP-1 binds directly to the hSHP promoter. Site-directed mutagenesis made it clear that the hSHP promoter activation by SREBP-1 is mostly mediated by the SRE1 (-186 to -195 bp) in the hSHP promoter, which is not conserved in the mouse SHP promoter. Moreover, adenovirus-mediated overexpression of SREBP-1c/ADD-1 induced SHP mRNA expression and repressed CYP7A1 expression in HepG2 cells. Finally, we found that a four-nucleotide deletion (-195CT-GAdel) in the hSHP promoter, which is reported to be associated with altered body weight and insulin secretion in human, coincides with the SRE1. This mutation strongly decreased both basal and SREBP-1 dependent activities of the hSHP promoter, because of the reduced binding of SREBP-1 to the mutated SRE1. Overall, our results demonstrate a differential regulation of human and mouse SHP promoters by SREBP-1. We propose a possible role of SREBP-1 in the species differential regulation of cholesterol and bile acid homeostasis via a novel mechanism of up-regulation of the hSHP gene expression.
Small heterodimer partner (SHP; NR0B2) is an atypical orphan nuclear receptor that lacks a conventional DNA binding domain (DBD) and represses the transcriptional activity of various nuclear receptors. In this study, we examined the novel cross talk between SHP and BETA2/NeuroD, a basic helix-loop-helix transcription factor. In vitro and in vivo protein interaction studies showed that SHP physically interacts with BETA2/NeuroD, but not its heterodimer partner E47. Moreover, confocal microscopic study and immunostaining results demonstrated that SHP colocalized with BETA2 in islets of mouse pancreas. SHP inhibited BETA2/NeuroD-dependent transactivation of an E-box reporter, whereas SHP was unable to repress the E47-mediated transactivation and the E-box mutant reporter activity. In addition, SHP repressed the BETA2-dependent activity of glucokinase and cyclin-dependent kinase inhibitor p21 gene promoters. Gel shift and in vitro protein competition assays indicated that SHP inhibits neither dimerization nor DNA binding of BETA2 and E47. Rather, SHP directly repressed BETA2 transcriptional activity and p300-enhanced BETA2/NeuroD transcriptional activity by inhibiting interaction between BETA2 and coactivator p300. We also showed that C-terminal repression domain within SHP is also required for BETA2 repression. However, inhibition of BETA2 activity was not observed by naturally occurring human SHP mutants that cannot interact with BETA2/NeuroD. Taken together, these results suggest that SHP acts as a novel corepressor for basic helix-loop-helix transcription factor BETA2/NeuroD by competing with coactivator p300 for binding to BETA2/NeuroD and by its direct transcriptional repression function.
The orphan nuclear receptor small heterodimer partner (SHP; NR0B2) is an unusual orphan nuclear receptor that lacks a conventional DNA-binding domain and acts as a modulator of transcriptional activities of a number of nuclear receptors. We have previously reported that the orphan nuclear receptor ERRgamma activates the SHP promoter. In this study, we have found that basic helix-loop-helix (bHLH) transcription factors, the E2A proteins (E47, E12 and E2/5), activated the human but not the mouse SHP promoter. In contrast, the tissue-specific E47 heterodimer partner BETA2 repressed the E47- mediated transactivation of the human SHP (hSHP) promoter. Using serial deletions and E-box mutant constructs of the hSHP promoter, we identified two E-boxes (E6 and E7) as E47-responsive E-boxes, which are not conserved in the mouse SHP promoter. Moreover, gel shift, chromatin immunoprecipitation (ChIP) and northern blot assays demonstrated that E47 directly binds to the hSHP promoter in vivo and in vitro and that Id proteins inhibited E47 binding to the hSHP promoter. Finally, we found that E47 and steroidogenic factor 1 (SF-1), a regulator of the SHP promoter, synergistically activate the human but not the mouse SHP promoter. Our findings suggest that the E2A proteins differentially regulate the human and mouse SHP promoters and cooperate with orphan nuclear receptor SF-1 for transcriptional activation of the hSHP promoter.
Cholesterol gallstone formation is a complex genetic trait. To identify additional cholesterol gallstone susceptibility loci, we performed a quantitative trait locus analysis using an intercross of PERA/Ei and I/LnJ inbred strains of mice.
Mice of both sexes were examined for gallstone weight and evaluated according to a scoring system for the physical chemistry of cholelithiasis during feeding of a lithogenic diet. Intercross offspring were genotyped, and linkage analysis was performed by interval mapping. Differences in messenger RNA expression of positional candidate genes were determined using reverse-transcription and real-time polymerase chain reaction.
We identified significant loci associated with gallstone weight on chromosomes 10 and 4, named Lith7 and Lith8, respectively (both susceptibility alleles conferred by strain I/LnJ). Positional candidate genes with higher expression in I/LnJ mice are Fxr (official symbol, Nr1h4), encoding the nuclear bile salt receptor, on chromosome 10 and Shp1 (official symbol, Nr0b2), encoding the small heterodimer partner 1, on chromosome 4. A significant locus associated with gallstone score on chromosome 17, named Lith9 (susceptibility allele conferred by strain PERA/Ei), colocalizes with the genes Abcg5 and Abcg8 that encode the canalicular cholesterol transporter. Higher hepatic messenger RNA expression of Abcg5 and Abcg8 in strain PERA/Ei correlates positively with higher biliary cholesterol levels.
Our findings suggest a primary role of the nuclear bile salt receptor FXR and the canalicular cholesterol transporter ABCG5/ABCG8 in the genetic susceptibility and pathogenesis of cholesterol cholelithiasis in these strains of inbred mice.
The bile acid-activated farnesoid X receptor (FXR) is a nuclear receptor regulating bile acid, glucose and cholesterol homeostasis. Obeticholic acid (OCA), a promising drug for the treatment of non-alcoholic steatohepatitis (NASH) and type 2 diabetes, activates FXR. Mouse studies demonstrated that FXR activation by OCA alters hepatic expression of many genes. However, no data are available on the effects of OCA in the human liver. Here we generated gene expression profiles in human precision cut liver slices (hPCLS) after treatment with OCA.
hPCLS were incubated with OCA for 24 h. Wild-type or FXR(-/-) mice received OCA or vehicle by oral gavage for 7 days.
Transcriptomic analysis showed that well-known FXR target genes, including NR0B2 (SHP), ABCB11 (BSEP), SLC51A (OSTα) and SLC51B (OSTβ), and ABCB4 (MDR3) are regulated by OCA in hPCLS. Ingenuity pathway analysis confirmed that 'FXR/RXR activation' is the most significantly changed pathway upon OCA treatment. Comparison of gene expression profiles in hPCLS and mouse livers identified 18 common potential FXR targets. ChIP-sequencing in mouse liver confirmed FXR binding to IR1 sequences of Akap13, Cgnl1, Dyrk3, Pdia5, Ppp1r3b and Tbx6.
Our study shows that hPCLS respond to OCA treatment by upregulating well-known FXR target genes, demonstrating its suitability to study FXR-mediated gene regulation. We identified six novel bona-fide FXR target genes in both mouse and human liver. Finally, we discuss a possible explanation for changes in high or low density lipoprotein observed in NASH and primary biliary cholangitis patients treated with OCA based on the genomic expression profile in hPCLS.
Transdifferentiattion potential of mesenchymal stem cells (MSCs) into insulin-producing cells (IPCs) has been suggested recently. In our recent works, we demonstrated the high performance of mouse neonate pancreas extract (MPE) in the production of functional IPCs from carcinoma stem cells. In this study, MPE was used to generate IPCs from MSCs without any genetic manipulation. To this end, bone marrow MSCs were isolated and characterized. In order to differentiate, MSCs were induced by selection of nestin-expressing cells and treatment with 100 μg/mL MPE. Morphological features of the differenti-ated cells were confirmed by dithizone staining. Immunoreactivity to insulin receptor beta, proinsulin, insulin, and C-peptide was observed by immunoflourescence. We also quantified glucose-dependent insulin production and secretion by ELISA. Real-time PCR indicated the expressions of β cell-related genes, PDX-1, INS1, INS2, EP300, and CREB1, in IPC cells. Possible pathways governed by CREB1, EP300, and PDX-1 transcription factors in differentiation of MSCs to IPCs were determined based on Gene Set Enrichment (GSE) approach at P = 0.05. Pathway discovery highlighted the negative regulatory effects of MIR124-2, HDAC5 protein, REST, and NR0B2 transcription factors on expression of CREB1, EP300, and PDX-1 and inhabitation of IPC differentiations. In contrast, a crosstalk between FOXA2 and TCF7L2 transcription factors, DNA-PK complex, KAT2B protein positively interacting with PDX-1, CREB1, EP300 resulted in the induction of IPC and following insulin production. In conclusion, we report an efficient, simple, and easy method for production of functional IPCs from MSCs by MPE treatment.
Disrupted bile secretion leads to liver damage characterized by inflammation, fibrosis, eventually cirrhosis, and hepatocellular cancer. As obstructive cholestasis often progresses insidiously, markers for the diagnosis and staging of the disease are urgently needed. To this end, we compiled a comprehensive data set of serum markers, histological parameters and transcript profiles at 8 time points of disease progression after bile duct ligation (BDL) in mice, aiming at identifying a set of parameters that could be used as robust biomarkers for transition of different disease progression phases.
Statistical analysis of the more than 6,000 data points revealed distinct temporal phases of disease. Time course correlation analysis of biochemical, histochemical and mRNA transcript parameters (=factors) defined 6 clusters for different phases of disease progression. The number of CTGF-positive cells provided the most reliable overall measure for disease progression at histological level, bilirubin at biochemical level, and metalloproteinase inhibitor 1 (Timp1) at transcript level. Prominent molecular events exhibited by strong transcript peaks are found for the transcriptional regulator Nr0b2 (Shp) and 1,25-dihydroxyvitamin D(3) 24-hydroxylase (Cyp24a1) at 6 h. Based on these clusters, we constructed a decision tree of factor combinations potentially useful as markers for different time intervals of disease progression. Best prediction for onset of disease is achieved by fibronectin (Fn1), for early disease phase by Cytochrome P450 1A2 (Cyp1a2), passage to perpetuation phase by collagen1α-1 (Col1a1), and transition to the progression phase by interleukin 17-a (Il17a), with early and late progression separated by Col1a1. Notably, these predictions remained stable even for randomly chosen small sub-sets of factors selected from the clusters.
Our detailed time-resolved explorative study of liver homogenates following BDL revealed a well-coordinated response, resulting in disease phase dependent parameter modulations at morphological, biochemical, metabolic and gene expression levels. Interestingly, a small set of selected parameters can be used as diagnostic markers to predict disease stages in mice with cholestatic liver disease.
Lung cancer is the leading cause of cancer-mediated death. Although various therapeutic approaches are used for lung cancer treatment, these mainly target the tumor suppressor p53 transcription factor, which is involved in apoptosis and cell cycle arrest. However, p53-targeted therapies have limited application in lung cancer, since p53 is found to be mutated in more than half of lung cancers. In this study, we propose tumor suppressor FOXA2 as an alternative target protein for therapies against lung cancer and reveal a possible FOXA2-centered transcriptional regulation network by identifying new target genes and binding partners of FOXA2 by using various screening techniques. The genes encoding Glu/Asp-rich carboxy-terminal domain 2 (CITED2), nuclear receptor subfamily 0, group B, member 2 (NR0B2), cell adhesion molecule 1 (CADM1) and BCL2-associated X protein (BAX) were identified as putative target genes of FOXA2. Additionally, the proteins including highly similar to heat shock protein HSP 90-beta (HSP90A), heat shock 70 kDa protein 1A variant (HSPA1A), histone deacetylase 1 (HDAC1) and HDAC3 were identified as novel interacting partners of FOXA2. Moreover, we showed that FOXA2-dependent promoter activation of BAX and p21 genes is significantly reduced via physical interactions between the identified binding partners and FOXA2. These results provide opportunities to understand the FOXA2-centered transcriptional regulation network and novel therapeutic targets to modulate this network in p53-deficient lung cancer. 
Testosterone production by Leydig cells is a tightly regulated process requiring synchronized expression of several steroidogenic genes by numerous transcription factors. Myocyte enhancer factor 2 (MEF2) are transcription factors recently identified in somatic cells of the male gonad. In other tissues, MEF2 factors are essential regulators of organogenesis and cell differentiation. So far in the testis, MEF2 factors were found to regulate Leydig cell steroidogenesis by controlling Nr4a1 and Star gene expression. To expand our understanding of the role of MEF2 in Leydig cells, we performed microarray analyses of MEF2-depleted MA-10 Leydig cells, and the results were analyzed using Partek and Ingenuity Pathway Analysis software. Several genes were differentially expressed in MEF2-depleted Leydig cells, and 16 were validated by quantitative RT-PCR. A large number of these genes are known to be involved in fertility, gonad morphology, and steroidogenesis. These include Ahr, Bmal1, Cyp1b1, Hsd3b1, Hsd17b7, Map2k1, Nr0b2, Pde8a, Por, Smad4, Star, and Tsc22d3, which were all downregulated in the absence of MEF2. In silico analyses revealed the presence of MEF2-binding sites within the first 2 kb upstream of the transcription start site of the Por, Bmal1, and Nr0b2 promoters, suggesting direct regulation by MEF2. Using transient transfections in MA-10 Leydig cells, small interfering RNA knockdown, and a MEF2-Engrailed dominant negative, we found that MEF2 activates the Por, Bmal1, and Nr0b2 promoters and that this requires an intact MEF2 element. Our results identify novel target genes for MEF2 and define MEF2 as an important regulator of Leydig cell function and male reproduction. 
The increasing incidence of male reproductive impairments has been associated with di-(2-ethylhexyl)-phthalate (DEHP) exposure. However, mechanisms involved are lacking. We exposed 4-week-old male C57BL/6j mice to DEHP by gavage at 0, 125, 250 or 500 mg/kg body weight/day for 28 consecutive days. Our data showed that pubertal exposure to DEHP induces sperm count reduction as well as histological abnormalities in seminiferous epithelium and apoptosis of post-meiotic germ cells, and these effects are concomitant with reduction of testosterone levels and its steroidogenic gene expression. Moreover, the expressions of estrogen receptor ERβ and nuclear receptors Nr0b1, Nr0b2 are increased. The expression of Nr5a2 which is the inducer of steroidogenesis is significantly reduced. Furthermore, spermatogonial stem cell (SSC) self-renewal, differentiation and meiosis were significantly impaired, and the epigenetic regulator G9a-mediated histone methylation was decreased following DEHP exposure. Our results suggest that the DEHP-induced male reproductive impairments may depend on its estrogenic action on estrogen receptor and nuclear receptor, and involve inhibition of steroidogenesis, SSC self-renewal and meiosis, which may be attributed to the down-regulation of G9a-mediated histone methylation.
We explored the molecular mechanisms of obesity and insulin resistance in patients with polycystic ovary syndrome (PCOS) using a human embryonic stem cell model (hESCs). Three PCOS-derived and one non-PCOS-derived hESC lines were induced into adipocytes, and then total RNA was extracted. The differentially expressed PCOS-derived and non-PCOS-derived adipocytes genes were identified using the Boao Biological human V 2.0 whole genome oligonucleotide microarray. Signals of interest were then validated by real-time PCR. A total of 153 differential genes were expressed of which 91 genes were up-regulated and 62 down-regulated. Nuclear receptor subfamily 0, group B, member 2 (NR0B2) was an up-regulated gene, and the GeneChip CapitalBio® Molecule Annotation System V4.0 indicated that it was associated with obesity and diabetes (Ratio ≥ 2.0X). Multiple genes are involved in PCOS. Nuclear receptor subfamily 0, group B, member 2 may play a role in obesity and insulin resistance in patients with PCOS.
The nuclear orphan receptors for which endogenous ligands have not been identified include nuclear receptor (NR)0B1 (adrenal hypoplasia congenita critical region on chromosome X gene), NR0B2 (small heterodimer partner), NR1D1/2 (Rev-Erbα/β), NR2C1 (testicular receptor 2), NR2C2 (testicular receptor 4), NR2E1 (tailless), NR2E3 (photoreceptor-specific NR [PNR]), NR2F1 chicken ovalbumin upstream promoter transcription factor 1 (COUP-TFI), NR2F2 (COUP-TFII), NR2F6 (v-erbA-related protein), NR4A1 (Nur77), NR4A2 (Nurr1), NR4A3 (Nor1), and NR6A1 (GCNF). These receptors play essential roles in development, cellular homeostasis, and disease including cancer where over- or underexpression of some receptors has prognostic significance for patient survival. Results of receptor knockdown or overexpression in vivo and in cancer cell lines demonstrate that orphan receptors exhibit tumor-specific pro-oncogenic or tumor suppressor-like activity. For example, COUP-TFII expression is both a positive (ovarian) and negative (prostate and breast) prognostic factor for cancer patients; in contrast, the prognostic activity of adrenal hypoplasia congenita critical region on chromosome X gene for the same tumors is the inverse of COUP-TFII. Functional studies show that Nur77 is tumor suppressor like in acute leukemia, whereas silencing Nur77 in pancreatic, colon, lung, lymphoma, melanoma, cervical, ovarian, gastric, and some breast cancer cell lines induces one or more of several responses including growth inhibition and decreased survival, migration, and invasion. Although endogenous ligands for the orphan receptors have not been identified, there is increasing evidence that different structural classes of compounds activate, inactivate, and directly bind several orphan receptors. Thus, the screening and development of selective orphan receptor modulators will have important clinical applications as novel mechanism-based agents for treating cancer patients overexpressing one or more orphan receptors and also for combined drug therapies. 
The molecular mechanisms behind human liver disease progression to cirrhosis remain elusive. Nuclear receptor small heterodimer partner (SHP/Nr0b2) is a hepatic tumor suppressor and a critical regulator of liver function. SHP expression is diminished in human cirrhotic livers, suggesting a regulatory role in human liver diseases. The goal of this study was to identify novel SHP-regulated genes that are involved in the development and progression of chronic liver disease. To achieve this, we conducted the first comprehensive RNA sequencing (RNA-seq) analysis of Shp(-/-) mice, compared the results with human hepatitis C cirrhosis RNA-seq and nonalcoholic steatohepatitis (NASH) microarray datasets, and verified novel results in human liver biospecimens. This approach revealed new gene signatures associated with chronic liver disease and regulated by SHP. Several genes were selected for validation of physiological relevance based on their marked upregulation, novelty with regard to liver function, and involvement in gene pathways related to liver disease. These genes include peptidoglycan recognition protein 2, dual specific phosphatase-4, tetraspanin 4, thrombospondin 1, and SPARC-related modular calcium binding protein-2, which were validated by qPCR analysis of 126 human liver specimens, including steatosis, fibrosis, and NASH, alcohol and hepatitis C cirrhosis, and in mouse models of liver inflammation and injury. This RNA-seq analysis identifies new genes that are regulated by the nuclear receptor SHP and implicated in the molecular pathogenesis of human chronic liver diseases. The results provide valuable transcriptome information for characterizing mechanisms of these diseases. 
The liver helps maintain energy homeostasis by synthesizing and storing glucose and lipids. Gonadal steroids, particularly estrogens, play an important role in regulating metabolism. As estrogens are considered female hormones, metabolic disorders related to the disruption of estrogen signaling have mostly been studied in females. Estrogen receptor alpha (ESR1) is the predominant receptor in both the male and female liver, and it mediates the hepatic response to estrogens. Loss of ESR1 increases weight gain and obesity in female rats, while reducing the normal growth in males. Although Esr1-/- male rats have a reduced body weight, they exhibit increased adipose deposition and impaired glucose tolerance. We further investigated whether these metabolic disorders in Esr1-/- male rats were linked with the loss of transcriptional regulation by ESR1 in the liver. To identify the ESR-regulated genes, RNA-sequencing was performed on liver mRNAs from wildtype and Esr1-/- male rats. Based on an absolute fold change of ≥2 with a p-value ≤ 0.05, a total of 706 differentially expressed genes were identified in the Esr1-/- male liver: 478 downregulated, and 228 upregulated. Pathway analyses demonstrate that the differentially expressed genes include transcriptional regulators (Cry1, Nr1d1, Nr0b2), transporters (Slc1a2), and regulators of biosynthesis (Cyp7b1, Cyp8b1), and hormone metabolism (Hsd17b2, Sult1e1). Many of these genes are also integral parts of the lipid and carbohydrate metabolism pathways in the liver. Interestingly, certain critical regulators of the metabolic pathways displayed a sexual dimorphism in expression, which may explain the divergent weight gain in Esr1-/- male and female rats despite common metabolic dysfunctions.
Glucocorticoids impair testosterone synthesis by an unknown mechanism. Stallions treated with the synthetic glucocorticoid dexamethasone had testes collected at 6 or 12 hours postinjection. The testicular expression of selected genes encoding nuclear receptors and steroidogenic enzymes was measured. At 6 hours, dexamethasone treatment decreased levels of NR0B2, NR4A1, NR5A1, and NR5A2 messenger RNAs (mRNAs) and NR5A2 mRNA levels remained depressed at 12 hours. In contrast, dexamethasone increased levels of NFKBIA mRNA at both time points. At 6 hours, dexamethasone did not alter levels of NR0B1, NR2F1, NR2F2, NR3C1, CYP11A1, CYP17A1, CYP19A1, DHCR24, GSTA3, HSD3B2, HSD17B3, LHCGR, or STAR mRNAs. In primary cultures of Leydig cells, 10 -9 and 10 -7 M dexamethasone decreased levels of NR4A1 and NR5A1 mRNAs and increased those of NFKBIA mRNA. Our discovery that dexamethasone downregulates NR4A1, NR5A1, and NR5A2 genes, known to be important for testicular functions, may be part of the mechanism by which glucocorticoids acutely decreases testosterone.
Inflammatory reaction plays a crucial role in the pathophysiology of acquired hearing loss such as ototoxicity and labyrinthitis. In our earlier work, we showed the pivotal role of otic fibrocytes in cochlear inflammation and the critical involvement of proinflammatory cytokines in cisplatin ototoxicity. We also demonstrated that otic fibrocytes inhibit monocyte chemoattractant protein 1 (CCL2) upregulation in response to interleukin-10 (IL-10) via heme oxygenase 1 (HMOX1) signaling, resulting in suppression of cochlear inflammation. However, it is still unclear how IL-10 affects inflammation-mediated cochlear injury. Here we aim to determine how hypochlorous acid, a model inflammation mediator affects cochlear cell viability and how IL-10 affects hypochlorous acid-mediated cochlear cell injury. NaOCl, a sodium salt of hypochlorous acid (HOCl) was found to induce cytotoxicity of HEI-OC1 cells in a dose-dependent manner. Combination of hydrogen peroxide and myeloperoxidase augmented cisplatin cytotoxicity, and this synergism was inhibited by N-Acetyl-L-cysteine and ML-171. The rat spiral ligament cell line (RSL) appeared to upregulate the antioxidant response element (ARE) activities upon exposure to IL-10. RSL cells upregulated the expression of NRF2 (an ARE ligand) and NR0B2 in response to CoPP (a HMOX1 inducer), but not to ZnPP (a HMOX1 inhibitor). Adenovirus-mediated overexpression of NR0B2 was found to suppress CCL2 upregulation. IL-10-positive cells appeared in the mouse stria vascularis 1 day after intraperitoneal injection of lipopolysaccharide (LPS). Five days after injection, IL-10-positive cells were observed in the spiral ligament, spiral limbus, spiral ganglia, and suprastrial area, but not in the stria vascularis. IL-10R1 appeared to be expressed in the mouse organ of Corti as well as HEI-OC1 cells. HEI-OC1 cells upregulated Bcl-xL expression in response to IL-10, and IL-10 was shown to attenuate NaOCl-induced cytotoxicity. In addition, HEI-OC1 cells upregulated IL-22RA upon exposure to cisplatin, and NaOCl cytotoxicity was inhibited by IL-22. Taken together, our findings suggest that hypochlorous acid is involved in cochlear injury and that IL-10 potentially reduces cochlear injury through not only inhibition of inflammation but also enhancement of cochlear cell viability. Further studies are needed to determine immunological characteristics of intracochlear IL-10-positive cells and elucidate molecular mechanisms involved in the otoprotective activity of IL-10.
Mammalian nuclear receptors (NRs) are transcription factors regulating the expression of target genes that play an important role in drug metabolism, transport, and cellular signaling pathways. The orphan and structurally unique receptor small heterodimer partner 1 (syn NR0B2) is not only known for its modulation of drug response, but has also been reported to be involved in hepatocellular carcinogenesis. Indeed, previous studies show that NR0B2 is downregulated in human hepatocellular carcinoma, suggesting that NR0B2 acts as a tumor suppressor via inhibition of cellular growth and activation of apoptosis in this tumor entity. The aim of our study was to elucidate whether NR0B2 may also play a role in other tumor entities. Comparing NR0B2 expression in renal cell carcinoma and adjacent nonmalignant transformed tissue revealed significant downregulation in vivo. Additionally, the impact of heterologous expression of NR0B2 on cell cycle progression and proliferation in cells of renal origin was characterized. Monitoring fluorescence intensity of resazurin turnover in RCC-EW cells revealed no significant differences in metabolic activity in the presence of NR0B2. However, there was a significant decrease of cellular proliferation in cells overexpressing this NR, and NR0B2 was more efficient than currently used antiproliferative agents. Furthermore, flow cytometry analysis showed that heterologous overexpression of NR0B2 significantly reduced the amount of cells passing the G1 phase, while on the other hand, more cells in S/G2 phase were detected. Taken together, our data suggest that downregulation of NR0B2 may also play a role in renal cell carcinoma development and progression. 
Residual feed intake (RFI) is a powerful indicator for energy utilization efficiency and responds to selection. Low RFI selection enables a reduction in feed intake without affecting growth performance. However, the effective variants or major genes dedicated to phenotypic differences in RFI in quality chickens are unclear. Therefore, a genome-wide association study (GWAS) and RNA sequencing were performed on RFI to identify genetic variants and potential candidate genes associated with energy improvement.
A lower average daily feed intake was found in low-RFI birds compared to high-RFI birds. The heritability of RFI measured from 44 to 83 d of age was 0.35. GWAS showed that 32 of the significant single nucleotide polymorphisms (SNPs) associated with the RFI (P &lt; 10(-4)) accounted for 53.01 % of the additive genetic variance. More than half of the effective SNPs were located in a 1 Mb region (16.3-17.3 Mb) of chicken (Gallus gallus) chromosome (GGA) 12. Thus, focusing on this region should enable a deeper understanding of energy utilization. RNA sequencing was performed to profile the liver transcriptomes of four male chickens selected from the high and low tails of the RFI. One hundred and sixteen unique genes were identified as differentially expressed genes (DEGs). Some of these genes were relevant to appetite, cell activities, and fat metabolism, such as CCKAR, HSP90B1, and PCK1. Some potential genes within the 500 Kb flanking region of the significant RFI-related SNPs detected in GWAS (i.e., MGP, HIST1H110, HIST1H2A4L3, OC3, NR0B2, PER2, ST6GALNAC2, and G0S2) were also identified as DEGs in chickens with divergent RFIs.
The GWAS findings showed that the 1 Mb narrow region of GGA12 should be important because it contained genes involved in energy-consuming processes, such as lipogenesis, social behavior, and immunity. Similar results were obtained in the transcriptome sequencing experiments. In general, low-RFI birds seemed to optimize energy employment by reducing energy expenditure in cell activities, immune responses, and physical activity compared to eating.
Breast-cancer is heterogeneous and consists of various groups with different biological characteristics. Innovative pharmacological approaches accounting for this heterogeneity are needed. The forty eight human Nuclear-Hormone-Receptors are ligand-dependent transcription-factors and are classified into Endocrine-Receptors, Adopted-Orphan-Receptors (Lipid-sensors and Enigmatic-Orphans) and Orphan-receptors. Nuclear-Receptors represent ideal targets for the design/synthesis of pharmacological ligands. We provide an overview of the literature available on the expression and potential role played by Lipid-sensors, Enigmatic-Orphans and Orphan-Receptors in breast-cancer. The data are complemented by an analysis of the expression levels of each selected Nuclear-Receptor in the PAM50 breast-cancer groups, following re-elaboration of the data publicly available. The major aim is to support the idea that some of the Nuclear-Receptors represent largely unexploited therapeutic-targets in breast-cancer treatment/chemo-prevention. On the basis of our analysis, we conclude that the Lipid-Sensors, NR1C3, NR1H2 and NR1H3 are likely to be onco-suppressors in breast-cancer. The Enigmatic-Orphans, NR1F1 NR2A1 and NR3B3 as well as the Orphan-Receptors, NR0B1, NR0B2, NR1D1, NR2F1, NR2F2 and NR4A3 exert a similar action. These Nuclear-Receptors represent candidates for the development of therapeutic strategies aimed at increasing their expression or activating them in tumor cells. The group of Nuclear-Receptors endowed with potential oncogenic properties consists of the Lipid-Sensors, NR1C2 and NR1I2, the Enigmatic-Orphans, NR1F3, NR3B1 and NR5A2, as well as the Orphan-Receptors, NR2E1, NR2E3 and NR6A1. These oncogenic Nuclear-Receptors should be targeted with selective antagonists, reverse-agonists or agents/strategies capable of reducing their expression in breast-cancer cells.
The orphan receptor small heterodimer partner (SHP, NR0B2) modulates the transcription activity of the MODY1 gene HNF4a. Mutations in SHP were found in 7% of Japanese obese young-onset type 2 diabetic patients and were associated with moderate obesity and increased birth weight. We investigated SHP in 1927 U.K. subjects, examining relationships with type 2 diabetes, obesity, and birth weight. Sequencing of the coding region of SHP in 122 obese, young-onset type 2 diabetic patients detected the polymorphism G171A. The polymorphism was not associated with diabetes in case control or familial association studies. The A allele (frequency 0.07) was not associated with obesity in type 2 diabetic subjects (n = 348), their parents (n = 272), or young nondiabetic adults (n = 925). However, the rare (&lt;1%) AA homozygotes had a raised BMI in each cohort; this was significant when all cohorts were combined (Z score = 0.67 AA vs. -0.05 G/x, P = 0.02). There was no association with corrected birth weight in 382 normal babies, but the only AA baby was 4,069 g. Our study suggests that genetic variation in SHP is unlikely to be common in the predisposition to diabetes, obesity, or increased birth weight in U.K. Caucasians.
The small heterodimer partner SHP (NR0B2) is an unusual nuclear receptor that lacks the typical DNA binding domain common to most nuclear receptors. SHP has been reported to act as a corepressor for several nuclear receptors, but its exact mechanism of action is still elusive. Here we show that SHP can interact with the liver X receptors LXRalpha (NR1H3) and LXRbeta (NR1H2), as demonstrated by glutathione-S-transferase pull-down assays, mammalian two-hybrid, and coimmunoprecipitation experiments. In transfection assays, SHP inhibits the expression of an artificial reporter driven by an LXR-response element and represses the transcriptional activation by LXR of the human ATP-binding cassette transporter 1 (ABCA1) promoter. Treatment of Caco-2 cells with bile acids, which activate farnesoid X receptor and subsequently induce SHP, leads to the repression of the human ABCG1 gene, an established LXR target gene. These results demonstrate that SHP is able to interact with LXR and to modulate its transcriptional activity.
The orphan nuclear receptor small heterodimer partner (SHP; NR0B2) interacts with a wide array of nuclear receptors and represses their transcriptional activity. SHP expression is regulated by several other members of the nuclear receptor superfamily, including the orphan receptors SF-1 and LRH-1, and the bile acid receptor FXR. We have found that the SHP promoter is also activated by the estrogen receptor-related receptor gamma (ERRgamma) but not the related ERRalpha and ERRbeta isoforms. SHP and ERRgamma mRNAs are coexpressed in several tissues, including pancreas, kidney, and heart, confirming the potential relevance of this transactivation. ERRgamma transactivation is dependent on only one of five previously characterized DNA-binding sites for SF-1, and this element differs from previously reported ERR response elements. However, treatment with the histone deacetylase inhibitor trichostatin A significantly increased ERRalpha and ERRbeta activity on this element indicating that the lack of activity of ERRalpha and -beta may depend on their association with co-repressor in vivo. Furthermore, using protease sensitivity assays on DNA bound receptors it was demonstrated that DNA sequence of different response elements may cause allosteric modulation of ERR proteins, which in turn may be responsible for the differential activities of these receptors on different response elements. SHP inhibits ERRgamma transactivation and physically interacts with all three members of ERR subfamily, as demonstrated by both yeast two-hybrid and biochemical assays. As with other SHP targets, this interaction is dependent on the AF-2 coactivator-binding site of ERRgamma and the previously described N-terminal receptor interaction domain of SHP. Several recently described SHP mutations associated with moderate obesity in humans block the inhibition of ERRgamma activity. Overall, these results identify a new autoregulatory loop controlling SHP gene expression and significantly extend the potential functional roles of the three ERRs.
Small heterodimer partner (SHP, NR0B2) is an atypical orphan nuclear receptor that inhibits transcriptional activation by several other nuclear receptors. We recently reported that mutations in the SHP gene are associated with insulin resistance. In the present study, we demonstrated that the SHP gene is expressed in adipose tissues. A reporter gene assay showed that a gene product of SHP increased the transcriptional activation of peroxisome proliferator-activated receptor (PPAR) gamma. SHP-mediated activation of PPARgamma was observed both in the presence and absence of the ligand of PPARgamma. Immunoprecipitation and glutathione S-transferase pull-down assay showed that SHP directly bound to PPARgamma and competed with nuclear receptor corepressor for binding to PPARgamma. Serial deletion studies indicated that the C terminus of SHP is important for PPARgamma activation. Mutant SHP proteins, which are found in naturally occurring mutation, showed less enhancing activity for PPARgamma than wild-type SHP. Our results suggest that SHP may act as an endogenous enhancer of PPARgamma.
Mutations in several genes encoding transcription factors of the hepatocyte nuclear factor (HNF) cascade are associated with maturity-onset diabetes of the young (MODY), a monogenic form of early-onset diabetes mellitus. The ability of the orphan nuclear receptor small heterodimer partner (SHP, NR0B2) to modulate the transcriptional activity of MODY1 protein, the nuclear receptor HNF-4alpha, suggested SHP as a candidate MODY gene. We screened 173 unrelated Japanese subjects with early-onset diabetes for mutations in this gene and found five different mutations (H53fsdel10, L98fsdel9insAC, R34X, A195S, and R213C) in 6 subjects as well as one apparent polymorphism (R216H), all present in the heterozygous state. Interestingly, all of the subjects with the mutations were mildly or moderately obese at onset of diabetes, and analysis of the lineages of these individuals indicated that the SHP mutations were associated with obesity rather than with diabetes. Therefore, an additional group of 101 unrelated nondiabetic subjects with early-onset obesity was screened for mutations in the SHP gene. Two of the previously observed mutations (R34X and A195S) and two additional mutations (R57W and G189E) were identified in 6 subjects, whereas no mutations were identified in 116 young nondiabetic lean controls (P = 0.0094). Functional studies of the mutant proteins show that the mutations result in the loss of SHP activity. These results suggest that genetic variation in the SHP gene contributes to increased body weight and reveal a pathway leading to this common metabolic disorder in Japanese.
Podocyte loss is one of the determining factors for the progression toward glomerulosclerosis. Podocyte is terminally differentiated and does not typically proliferate following injury and loss. However, recent evidence suggested that during renal injury, podocyte could re-enter the cell cycle, sensitizing the cells to injury and death, but the molecular mechanisms underlying it, as well as the cell fate determination still remained unclear. Here, using NPHS2 Cre; mT/mG transgenic mice and primary podocytes isolated from the mice, we investigated the effect of mammalian target of rapamycin complex 1 (mTORC1)/4E-binding protein 1 (4E-BP1) signaling pathway on cell cycle re-entry and apoptosis of podocyte induced by adriamycin. It was found that podocyte cell cycle re-entry could be induced by adriamycin as early as the 1st week in vivo and the 2nd hour in vitro, accompanied with 4E-BP1 activation and was followed by podocyte loss or apoptosis from the 4th week in vivo or the 4th hour in vitro. Importantly, targeting 4E-BP1 activation by the RNA interference of 4E-BP1 or pharmacologic rapamycin (inhibitor of mTORC1, blocking mTORC1-dependent phosphorylation of its substrate 4E-BP1) treatment was able to inhibit the increases of PCNA, Ki67, and the S-phase fraction of cell cycle in primary podocyte during 2-6 h of adriamycin treatment, and also attenuated the following apoptotic cell death of podocyte detected from the 4th hour, suggesting that 4E-BP1 could be a regulator to manipulate the amount of cell cycle re-entry provided by differentiated podocyte, and thus regulate the degree of podocyte apoptosis, bringing us a new potential podocyte-protective substance that can be used for therapy.
Cyclin-dependent protein kinase 6 (CDK6) plays a pivotal role in the regulation of the cell cycle and cell proliferation in mammals, and disruption of its expression by various microRNAs has been implicated in the pathogenesis of multiple human cancers. In mammals, miR-34a acts as a downstream effector of p53, and thus indirectly targets Cdk6, abrogating its effects. However, no studies have been done so far to examine the mechanistic involvement of miR-34a in the silencing of cdk6 in fish. In the present study, we found that the cDNA sequence of whitefish cdk6 has a 3'UTR region that contains a binding site for miR-34a. Using a luciferase reporter assay, we demonstrated that whitefish cdk6 is a direct target of miR-34a in vitro. In order to confirm this relationship in vivo, we measured the miR-34a and cdk6 mRNA expression patterns in the liver of whitefish after short-term (8, 24, and 48 h) and long-term (14 and 28 days) exposure to microcystin-LR (MC-LR), a known hepatotoxin and tumor promoter. In contrast to the in vitro findings, we noticed an up-regulation of miR-34a and cdk6 expression after long-term MC-LR treatment. While these results indicate that both, miR-34a and cdk6 are responsive to MC-LR treatment, they do not support the presence of a miR-34a:cdk6 mRNA regulatory pair in the MC-LR-challanged whitefish liver in vivo. On the other hand, our findings suggests that cell regulatory elements, partnering with either miR-34a or cdk6, are worthy of further screening to better understand the molecular mechanisms that underlie the physiological response of fish challenged with hepatotoxic environmental pollutants like microcystins.
The recently discovered circular RNAs (circRNAs) are mostly formed by back-splicing where the downstream 5' splice site splices to the upstream 3' splice site by conventional pre-mRNA splicing. These circRNAs regulate gene expression by acting as sponges for micro-RNAs or RNA-binding proteins. Here we show that the NR5A1 (previously called Ad4BP or SF-1) gene which is exclusively expressed in the adrenal cortex and steroidogenic tissue can form atypical circRNAs by unconventional splicing. Two stem loops with inositol-requiring protein-1α (IRE1α) cleavage sites are connected by an IRE1α cleavage site to form a circRNA (circIRE RNA). From total RNA of normal human adrenal cortex, we detected a circIRE RNA with connected ends by IRE1α cleavage sites in exon 6 and exon 1 (circIRE NR5A1 ex6-1 RNA). circIRE NR5A1 ex6-1 RNA was not detected in the adrenocortical cancer cell line, H295R. When IRE1α was expressed in H295R cells a different circIRE NR5A1 RNA connecting IRE1-cleavage sites in exon 7 and exon 1 was detected (circIRE NR5A1 ex7-1 RNA). The expression of this circIRE RNA was inhibited by the IRE1 inhibitor 1, STF-083010, implicating that it was formed via the ER stress pathway, where IRE1α is a major factor. This is the first report of this type of circular RNA connected by IRE1-cleavage sites found to be expressed in mammalian cells in a tissue-specific manner. To our surprise, the concomitant expression of NR5A1 was increased by IRE1α implicating that NR5A1 was not subjected to IRE1-dependent decay of mRNA (RIDD) but rather activating a transcriptional regulatory network to cope with ER stress in steroidogenic tissue reminiscent to XBP1 in other tissue. We believe this is the first report of such tissue-specific transcriptional cascade responding to ER stress as well as the novel finding of circular RNAs connected by IRE1α cleavage sites expressed in mammalian tissue.
Excessive or inappropriate inflammatory responses can cause serious and even fatal diseases. The CCAAT/enhancer-binding protein alpha (CEBPA) gene encodes C/EBPα, a transcription factor that plays a fundamental role in controlling maturation of the myeloid lineage and is also expressed during the late phase of inflammatory responses when signs of inflammation are decreasing. MTL-CEBPA, a small activating RNA targeting for upregulation of C/EBPα, is currently being evaluated in a phase 1b trial for treatment of hepatocellular carcinoma. After dosing, subjects had reduced levels of pro-inflammatory cytokines, and we therefore hypothesized that MTL-CEBPA has anti-inflammatory potential. The current study was conducted to determine the effects of C/EBPα saRNA - CEBPA-51 - on inflammation in vitro and in vivo after endotoxin challenge. CEBPA-51 led to increased expression of the C/EBPα gene and inhibition of pro-inflammatory cytokines in THP-1 monocytes previously stimulated by E. coli-derived lipopolysaccharide (LPS). Treatment with MTL-CEBPA in an LPS-challenged humanized mouse model upregulated C/EBPα mRNA, increased neutrophils, and attenuated production of several key pro-inflammatory cytokines, including TNF-α, IL-6, IL-1β, and IFN-γ. In addition, a Luminex analysis of mouse serum revealed that MTL-CEBPA reduced pro-inflammatory cytokines and increased the anti-inflammatory cytokine IL-10. Collectively, the data support further investigation of MTL-CEBPA in acute and chronic inflammatory diseases where this mechanism has pathogenic importance.
Objective To observe the effect of low-frequency pulsed electromagnetic fields(PEMFs) on bone formation in rat osteoblasts(ROBs) and explore the mechanism of action of the cyclic adenosine monophosphate(cAMP)/protein kinase A(PKA)/cyclic adenosine effect binding protein(CREB) signaling pathway.Methods The skulls of newborn Wistar rats were harvested,and the ROBs were obtained by multiple enzymatic digestion methods for subculture. After treatment with 50 Hz 0.6 mT PEMFs for 3,6,and 9 days,the alkaline phosphatase(ALP) concentration in ROBs was detected;after 0,15,30,60,90,and 120 min,the expression of bone formation-related factor(RUNX2),the protein expression of osteogenesis-associated transcription factor(OSX),the cAMP concentration,as well as the protein expressions of p-PKA,p-CREB,and CREB were detected. The p-CREB nuclear translocation was observed. After interference with IFT88 by RNA interference,the expressions of RUNX2,OSX,p-PKA,and p-CREB protein in ROBs were detected.Results After treatment with PEMFs for 3,6,and 9 days,the ALP activity values in ROBs were 24.356±4.911,37.688±2.151,and 39.922±5.486,respectively,which were significantly higher than 18.531±2.401(P=0.0121),33.675±4.366(P=0.0324),and 36.574±1.339(P=0.0134) in the control groups. RUNX2 and OSX activities in ROBs were significantly higher than untreated group after PEMFs treatment for 30(P=0.0042 and P=0.0058),60(P=0.0097 and P=0.0079),and 90 min(P=0.0083 and P=0.0098). After PEMFs treatment for 30(P=0.0012) and 60 min(P=0.0035),the cAMP concentrations in ROBs were significantly higher than that in untreated group. After PEMFs treatment for 15(P=0.0018),30(P=0.0087),90(P=0.0250),and 120 min(P=0.0350),the p-PKA levels in ROBs were significantly higher than that in the untreated group. After PEMFs treatment for 15(P=0.0075),30(P=0.0017),60(P=0.0074),and 90 min(P=0.0096),the level of p-CREB in the ROBs was significantly higher than in the untreated group. After PEMFs treatment of ROBs for 15 min,CREB phosphorylated and accumulated in the nuclei. PKA and p-PKA were co-localized with primary cilia and stained,and it was found that p-PKA was localized on the primary cilia. After the primary cilia was removed by RNA interference,the protein expression levels of p-PKA(F=78.602,P=0.0270),p-CREB(F=76.082,P=0.0089),RUNX2(F=41.064,P=0.0230) and OSX(F=57.524,P=0.0310) were significantly lower than those of the non-interfered group.Conclusion PEMFs promote bone formation in ROBs by activating the primary cilia-associated cAMP/PKA/CREB signaling pathway.
The lethal-7 (let-7) family members and their targets are involved in the development and progression of tumors. Let-7-related polymorphisms have been reported to be associated with tumorigenesis and prognosis. In gastric cancer, however, the related studies are limited.
To investigate the role of let-7-related microRNA polymorphisms in the tumorigenesis and prognosis of gastric cancer in a Chinese population.
A total of 898 gastric cancer patients and 992 tumor-free controls were recruited into this study from 2008 to 2013. Gastric cancer patients were followed periodically. Ten single nucleotide polymorphisms (SNPs) in the let-7 gene region or their target mRNAs were genotyped using the MassARRAY system and their associations with the risk for or overall survival of gastric cancer were analyzed.
All the ten SNPs were in Hardy-Weinberg equilibrium. The C allele of the rs3811463 polymorphism in the 3'-untranslated region (UTR) of LIN28A was associated with a lower risk of gastric cancer [odds ratio (OR) = 0.74, 95% confidence interval (CI): 0.61-0.88, P = 0.001] after adjustment for age and Helicobacter pylori status. Seven hundred and thirty-five gastric cancer patients who had undergone radical tumorectomy were included in the survival analysis and their 5-year survival rate was 53.9% (95%CI: 50.1%-57.6%). The rs10889677 in the 3'-UTR of IL23R was corresponded to the prognosis of gastric cancer in a dose-response manner, in which the death risk increased by 25% [hazard ratio (HR) = 1.25, 95%CI: 1.04-1.45, P = 0.011] with each increase in the number of C alleles after controlling for other potential clinicopathological parameters.
The let-7-related polymorphism rs3811463 in LIN28A is associated with the susceptibility to gastric cancer and the let-7-related polymorphism rs10889677 in IL23R is associated with the prognosis of gastric cancer.
Protein structural vibrations impact biology by steering the structure to functional intermediate states; enhancing tunneling events; and optimizing energy transfer. Strong water absorption and a broad continuous vibrational density of states have prevented optical identification of these vibrations. Recently spectroscopic signatures that change with functional state were measured using anisotropic terahertz microscopy. The technique however has complex sample positioning requirements and long measurement times, limiting access for the biomolecular community. Here we demonstrate that a simplified system increases spectroscopic structure to dynamically fingerprint biomacromolecules with a factor of 6 reduction in data acquisition time. Using this technique, polarization varying anisotropy terahertz microscopy, we show sensitivity to inhibitor binding and unique vibrational spectra for several proteins and an RNA G-quadruplex. The technique's sensitivity to anisotropic absorbance and birefringence provides rapid assessment of macromolecular dynamics that impact biology.
Aromatic antiepileptic drugs (AEDs) are frequently implicated in cutaneous adverse drug reactions (cADRs), a few of which are associated with certain human leukocyte antigen (HLA) alleles in some populations. We aimed to find HLA-associations with AED-related cADRs among North Indians.
North Indian subjects with cADR due to an AED, and those who were AED-tolerant were recruited as cases and controls, respectively. Genotyping for HLA-A, B and DRB1 were performed. Statistical analysis to compare carrier-rates and allele-frequencies between cases and controls (and healthy population, where necessary), was done for HLA-alleles occurring more than twice in either group.
120 cases {11 - Lamotrigine (LTG), 14 -Valproic acid (VPA), 8 -Levetiracetam (LEV), 35 -Carbamazepine (CBZ) and 52 - Phenytoin (PHT)}, and 250 controls were recruited. Presence of HLA-A*31:01 and HLA-B*51:01 were found to increase the risk of Maculopapular exanthema (MPE) due to CBZ and PHT (OR = 6.38; 95% CI: 1.46-27.75; OR = 4.60; 95% CI: 1.54-13.72, respectively). Among the severe cADRs, HLA-B*57:01(OR = 11.00 95% CI: 1.41-85.81) and HLA-DRB1*07:01 (OR = 7.25; 95% CI: 1.09-48.18) were noted to be significantly associated with CBZ-induced Stevens Johnson Syndrome (SJS)/Toxic Epidermal Necrolysis (TEN); HLA-B *51:01 was associated with drug reaction eosinophilia and systemic symptoms (DRESS) caused by PHT (OR = 6.90; 95% CI: 1.38-34.29).
We found significant associations of some HLA alleles with specific cADRs to CBZ and PHT in North Indians, which may need to be tested before AED-initiation; only screening for HLA-B*15:02 may not help in this population.
The aged kidney is susceptible to acute injury due presumably to its decreased ability to handle additional challenges, such as endoplasmic reticulum (ER) stress. This was tested by giving tunicamycin, an ER stress inducer, to either old or young mice. Injection of high dose caused renal failure in old mice, not in young mice. Moreover, injection of low dose resulted in severe renal damage in old mice, confirming the increased susceptibility of aged kidney to ER stress. There existed an abnormality in ER stress response kinetics in aged kidney, characterized by a loss of XBP-1 splicing and decreased PERK-eIF2α phosphorylation at late time point. The presence of excessive oxidative stress in aged kidney may play a role since high levels of oxidation increased ER stress-induced cell death and decreased IRE1 levels and XBP-1 splicing. Importantly, treatment with antioxidants protected old mice from kidney injury and normalized IRE1 and XBP-1 responses. Furthermore, older mice (6 months old) transgenic with antioxidative stress AGER1 were protected from ER stress-induced kidney injury. In conclusion, the decreased ability to handle ER stress, partly due to the presence of excessive oxidative stress, may contribute to increased susceptibility of the aging kidney to acute injury.
MicroRNAs (miRNAs) are post-transcriptional regulators of gene expression. Heterozygous loss-of-function point mutations of miRNA genes are associated with several human congenital disorders1-5, but neomorphic (gain-of-new-function) mutations in miRNAs due to nucleotide substitutions have not been reported. Here we describe a neomorphic seed region mutation in the chondrocyte-specific, super-enhancer-associated MIR140 gene encoding microRNA-140 (miR-140) in a novel autosomal dominant human skeletal dysplasia. Mice with the corresponding single nucleotide substitution show skeletal abnormalities similar to those of the patients but distinct from those of miR-140-null mice6. This mutant miRNA gene yields abundant mutant miR-140-5p expression without miRNA-processing defects. In chondrocytes, the mutation causes widespread derepression of wild-type miR-140-5p targets and repression of mutant miR-140-5p targets, indicating that the mutation produces both loss-of-function and gain-of-function effects. Furthermore, the mutant miR-140-5p seed competes with the conserved RNA-binding protein Ybx1 for overlapping binding sites. This finding may explain the potent target repression and robust in vivo effect by this mutant miRNA even in the absence of evolutionary selection of miRNA-target RNA interactions, which contributes to the strong regulatory effects of conserved miRNAs7,8. Our study presents the first case of a pathogenic gain-of-function miRNA mutation and provides molecular insight into neomorphic actions of emerging and/or mutant miRNAs.
β-Asarone (1,2,4-trimethoxy-5-[(Z)-prop-1-enyl]benzene) is an essential component of Acorus tatarinowii Schott volatile oil. Previous research has observed that β-asarone effectively attenuated symptoms in parkinsonian rats and improved their performance, but the mechanism of this effect remains unclear. Other research has shown that endoplasmic reticulum (ER) stress plays an important role in the pathogenesis of Parkinson's disease (PD). The protein kinase RNA-like endoplasmic reticulum kinase (PERK) was observed in the nigrostriatal dopaminergic neurons of patients with PD. However, our group observed that ER stress and autophagy occurred in 6-hydroxy dopamine (6-OHDA)-induced parkinsonian rats, and ER stress might induce autophagy. We assume that the protective role of β-asarone in parkinsonian rats is mediated via the ER stress-autophagy pathway. To support this hypothesis, we investigated the expressions of glucose regulated protein 78 (GRP78), PERK phosphorylation (p-PERK), C/EBP homologous binding protein (CHOP), Bcl-2 and Beclin-1 in 6-OHDA-induced parkinsonian rats after β-asarone treatment. The results showed that the β-asarone group and PERK inhibitor group had lower levels of GRP78, p-PERK, CHOP and Beclin-1 while having higher levels of Bcl-2. We deduced that β-asarone might regulate the ER stress-autophagy via inhibition of the PERK/CHOP/Bcl-2/Beclin-1 pathway in 6-OHDA-induced parkinsonian rats.
Innate immunity is a system that recognizes primarily and excludes pathogenic microorganism. MAVS/IPS-1/Cardif/Visa functions as an adapter protein for RIG-I like receptors (RLRs) and plays a key role in the production of antiviral proteins, interferons (IFNs), for RNA viruses. However, the activation mechanism is not fully understood. Here, we show that BinCARD isoform2 (BinCARD2), carrying CARD domain structure like MAVS, functions in innate immune response. Knockdown of BinCARD2 reduced the RLR ligand-induced expression of IFN-β mRNA and activation of the IFNB promoter. The activation of the IFNB promoter by overexpression of MAVS or TBK1 was suppressed by silencing of BinCARD2, but no effect on IFNB promoter activation by overexpression of TRIF or constitutive activated IRF-3. Furthermore, we confirmed that BinCARD2 protein associated with MAVS but not TBK1 by immunoprecipitation and colocalized with MAVS. Accordingly, we investigated whether BinCARD2 was involved in MAVS activation and showed that siBinCARD2 did not affect RIG-I/MAVS binding but impaired the MAVS oligomerization. Moreover, we infected A549 cells with vesicular stomatitis virus (VSV) and found that induction of IFN-β and IL-6 mRNA after VSV infection was decreased by BinCARD2 knockdown. Thus, these data may suggest that BinCARD2 associates with MAVS to positively modulate the oligomerization in the RIG-I like receptors pathway and activates innate immune response.
Transcriptomes consist of several classes of RNA that have wide-ranging but often poorly described functions and the deregulation of which leads to numerous diseases. Engineering of functionalized RNA-binding proteins (RBPs) could therefore have many applications. Our previous studies suggested that the RanBP2-type Zinc Finger (ZF) domain is a suitable scaffold to investigate the design of single-stranded RBPs. In the present work, we have analyzed the natural sequence specificity of various members of the RanBP2-type ZF family and characterized the interaction with their target RNA. Surprisingly, our data showed that natural RanBP2-type ZFs with different RNA-binding residues exhibit a similar sequence specificity and therefore no simple recognition code can be established. Despite this finding, different discriminative abilities were observed within the family. In addition, in order to target a long RNA sequence and therefore gain in specificity, we generated a 6-ZF array by combining ZFs from the RanBP2-type family but also from different families, in an effort to achieve a wider target sequence repertoire. We showed that this chimeric protein recognizes its target sequence (20 nucleotides), both in vitro and in living cells. Altogether, our results indicate that the use of ZFs in RBP design remains attractive even though engineering of specificity changes is challenging.
Multiple metabolic, immune and reproductive effects have been reported in fish residing in effluent-impacted sites. Natural stressors such as parasites also have been shown to impact the responses of organisms to chronic exposure to municipal effluent in the St. Lawrence River (Quebec, Canada). In order to comprehensively evaluate the cumulative impacts of anthropogenic and natural stressors on the health of yellow perch, differential mRNA transcription profiles were examined in juvenile females collected from effluent-impacted and upstream sites with low or high infection levels of the larval trematode Apophallus brevis. Transcriptomics was used to identify biological pathways associated with environmental exposure. In total, 3463 isoforms were differentially transcribed between sites. Patterns reflecting the combined effects of stressors were numerically dominant, with a majority of downregulated transcripts (68%). The differentially expressed transcripts were associated with 27 molecular and cellular functions ranging from cellular development to xenobiotic metabolism and were involved in the development and function of 13 organ systems including hematological, hepatic, nervous, reproductive and endocrine systems. Based on RNA-seq results, sixteen genes were measured by qPCR. Significant differences were observed for six genes in fish exposed to both stressors combined, whereas parasites and effluent individually impacted the transcription of one gene. Lysozyme activity, lipid peroxidation, retinol-binding protein and glucose-6-phosphate dehydrogenase were selected as potential biomarkers of effects to study specific pathways of interest. Lipid peroxidation in perch liver was different between sites, parasite loads, and for combined stressors. Overall, results indicated that juvenile yellow perch responded strongly to combined parasite and effluent exposure, suggesting cumulative effects on immune responses, inflammation and lipid metabolism mediated by retinoid receptors. The present study highlight the importance of using a comprehensive approach combining transcriptomics and endpoints measured at higher levels of biological organization to better understand cumulative risks of contaminants and pathogens in aquatic ecosystems.
Spinal muscular atrophy (SMA) is a progressive motor neuron disease caused by loss or mutation of the survival motor neuron 1 (SMN1) gene and retention of SMN2. We performed targeted capture and sequencing of the SMN2, CFTR, and PLS3 genes in 217 SMA patients. We identified a 6.3 kilobase deletion that occurred in both SMN1 and SMN2 (SMN1/2) and removed exons 7 and 8. The deletion junction was flanked by a 21 bp repeat that occurred 15 times in the SMN1/2 gene. We screened for its presence in 466 individuals with the known SMN1 and SMN2 copy numbers. In individuals with 1 SMN1 and 0 SMN2 copies, the deletion occurred in 63% of cases. We modeled the deletion junction frequency and determined that the deletion occurred in both SMN1 and SMN2. We have identified the first deletion junction where the deletion removes exons 7 and 8 of SMN1/2. As it occurred in SMN1, it is a pathogenic mutation. We called variants in the PLS3 and SMN2 genes, and tested for association with mild or severe exception patients. The variants A-44G, A-549G, and C-1897T in intron 6 of SMN2 were significantly associated with mild exception patients, but no PLS3 variants correlated with severity. The variants occurred in 14 out of 58 of our mild exception patients, indicating that mild exception patients with an intact SMN2 gene and without modifying variants occur. This sample set can be used in the association analysis of candidate genes outside of SMN2 that modify the SMA phenotype.
Regulation of pro-protein convertase subtilisin/kexin type 9 (PCSK9) by drugs has led to the development of a still small number of agents with powerful activity on low-density lipoprotein cholesterol levels, associated with a significant reduction of cardiovascular events in patients in secondary prevention. The Further Cardiovascular Outcomes Research with PCSK9 Inhibition in Subjects with Elevated Risk (FOURIER) and Evaluation of Cardiovascular Outcomes After an Acute Coronary Syndrome During Treatment With Alirocumab (ODYSSEY OUTCOMES) studies, with the two available PCSK9 antagonists, i.e. evolocumab and alirocumab, both reported a 15% reduction in major adverse cardiovascular events. Regulation of PCSK9 expression is dependent upon a number of factors, partly genetic and partly associated to a complex transcriptional system, mainly controlled by sterol regulatory element binding proteins. PCSK9 is further regulated by concomitant drug treatments, particularly by statins, enhancing PCSK9 secretion but decreasing its stimulatory phosphorylated form (S688). These complex transcriptional mechanisms lead to variable circulating levels making clinical measurements of plasma PCSK9 for cardiovascular risk assessment a debated matter. Determination of total PCSK9 levels may provide a diagnostic tool for explaining an apparent resistance to PCSK9 inhibitors, thus indicating the need for other approaches. Newer agents targeting PCSK9 are in clinical development with a major interest in those with a longer duration of action, e.g. RNA silencing, allowing optimal patient compliance. Interest has been expanded to areas not only limited to low-density lipoprotein cholesterol reduction but also investigating other non-lipid pathways raising cardiovascular risk, in particular inflammation associated to raised high-sensitivity C-reactive protein levels, not significantly affected by the present PCSK9 antagonists.
BACKGROUND The phosphatase actin regulator-1 (PHACTR-1) gene on chromosome 6 encodes an actin and protein phosphatase 1 (PP1) binding protein, Phactr-1, which is highly expressed in brain tissues. Phactr-1 expression is involved in physiological and pathological cerebral microvascular events. This study aimed to investigate the role of expression of Phactr-1 in a mouse brain capillary endothelial cell line, bEnd.3, by knockdown the PHACTR-1 gene. MATERIAL AND METHODS Three bEnd.3 cell groups were studied, CON (normal control cells), NC (control scramble transfected cells), and KD (cells with PHACTR-1 gene knockdown). The PHACTR-1 gene was knocked down using transfection with small hairpin RNA (shRNA). In the three cell groups cell proliferation, migration, and apoptosis were studied by MTT and colony formation assays, transwell and scratch assays, and flow cytometry. The related cell pathways of associated with Phactr-1 knockdown were studied by Western blot. RESULTS Phactr-1 knockdown suppressed bEnd.3 cell proliferation and migration, promoted cell apoptosis, and downregulated the expressions of migration-associated proteins, including matrix metalloproteinase (MMP)-2 and MMP-9 and upregulated apoptosis-associated proteins, including Bax, Bcl-2, cleaved caspase-3, and caspase-3. CONCLUSIONS Phactr-1 was shown to have a role in the inhibition of endothelial cell proliferation and migration, promoted cell apoptosis, and regulated matrix metalloproteinases and apoptosis-associated proteins. These findings indicate that the expression of the Phactr-1 should be studied further in the cerebral microvasculature, both in vitro and in vivo, regarding its potential as a diagnostic and therapeutic target for cerebral microvascular disease.
About one-third of dilated cardiomyopathy (DCM) cases are caused by mutations in sarcomere or cytoskeletal proteins. However, treating the cytoskeleton directly is not possible because drugs that bind to actin are not well tolerated. Mutations in the actin binding protein CAP2 can cause DCM and KO mice, either whole body (CAP2-KO) or cardiomyocyte-specific KOs (CAP2-CKO) develop DCM with cardiac conduction disease. RNA sequencing analysis of CAP2-KO hearts and isolated cardiomyocytes revealed overactivation of fetal genes, including serum response factor-regulated (SRF-regulated) genes such as Myl9 and Acta2 prior to the emergence of cardiac disease. To test if we could treat CAP2-KO mice, we synthesized and tested the SRF inhibitor CCG-1423-8u. CCG-1423-8u reduced expression of the SRF targets Myl9 and Acta2, as well as the biomarker of heart failure, Nppa. The median survival of CAP2-CKO mice was 98 days, while CCG-1423-8u-treated CKO mice survived for 116 days and also maintained normal cardiac function longer. These results suggest that some forms of sudden cardiac death and cardiac conduction disease are under cytoskeletal stress and that inhibiting signaling through SRF may benefit DCM by reducing cytoskeletal stress.
Clustered regularly interspaced short palindromic repeat (CRISPR) methodology is not only an efficient tool in gene editing but also an attractive platform to facilitate DNA, RNA, and protein interactions. We describe here the implementation of a CRISPR-based system to regulate expression in the clinically important yeast Candida albicans By fusing an allele of Streptococcus pyogenes Cas9 devoid of nuclease activity to a transcriptional repressor (Nrg1) or activator (Gal4), we were able to show specific repression or activation of the tester gene CAT1, encoding the cytosolic catalase. We generated strains where a 1.6-kbp upstream regulatory region of CAT1 controls the expression of the green fluorescent protein (GFP) and demonstrated the functionality of the constructs by quantitative PCR (qPCR), flow cytometry, and analysis of sensitivity/resistance to hydrogen peroxide. Activation and repression were strongly dependent on the position of the complex in this regulatory region. We also improved transcriptional activation using an RNA scaffolding strategy to allow interaction of inactive variants of Cas9 (dCas9) with the RNA binding protein MCP (monocyte chemoattractant protein) fused to the VP64 activator. The strategy shown here may facilitate the analysis of complex regulatory traits in this fungal pathogen.IMPORTANCE CRISPR technology is a new and efficient way to edit genomes, but it is also an appealing way to regulate gene expression. We have implemented CRISPR as a gene expression platform in Candida albicans using fusions between a Cas9 inactive enzyme and specific repressors or activators and demonstrated its functionality. This will allow future manipulation of complex virulence pathways in this important fungal pathogen.
Natural and synthetic small molecules from the NCI Developmental Therapeutics Program (DTP) were employed in molecular dynamics-based docking with DNA repair proteins whose RNA-Seq based expression was associated with overall cancer survival (OS) after adjustment for the PCNA metagene. The compounds employed were required to elicit a sensitive response (vs. resistance) in more than half of the cell lines tested for each cancer. Methodological approaches included peptide sequence alignments and homology modeling for 3D protein structure determination, ligand preparation, docking, toxicity and ADME prediction. Docking was performed for unique lists of DNA repair proteins which predict OS for AML, cancers of the breast, lung, colon, and ovaries, GBM, melanoma, and renal papillary cancer. Results indicate hundreds of drug-like and lead-like ligands with best-pose binding energies less than -6 kcal/mol. Ligand solubility for the top 20 drug-like hits approached lower bounds, while lipophilicity was acceptable. Most ligands were also blood-brain barrier permeable with high intestinal absorption rates. While the majority of ligands lacked positive prediction for HERG channel blockage and Ames carcinogenicity, there was a considerable variation for predicted fathead minnow, honey bee, and Tetrahymena pyriformis toxicity. The computational results suggest the potential for new targets and mechanisms of repair inhibition and can be directly employed for in vitro and in vivo confirmatory laboratory experiments to identify new targets of therapy for cancer survival.
The RNA-binding protein GRSF1 (G-rich RNA sequence-binding factor 1) critically maintains mitochondrial homeostasis. Accordingly, loss of GRSF1 impaired mitochondrial respiration and increased the levels of reactive oxygen species (ROS), triggering DNA damage, growth suppression, and a senescent phenotype characterized by elevated production and secretion of interleukin (IL)6. Here, we characterize the pathways that govern IL6 production in response to mitochondrial dysfunction in GRSF1-depleted cells. We report that loss of GRSF1 broadly altered protein expression programs, impairing the function of respiratory complexes I and IV. The rise in oxidative stress led to increased DNA damage and activation of mTOR, which in turn activated NF-κB to induce IL6 gene transcription and orchestrate a pro-inflammatory program. Collectively, our results indicate that GRSF1 helps preserve mitochondrial homeostasis, in turn preventing oxidative DNA damage and the activation of mTOR and NF-κB, and suppressing a transcriptional pro-inflammatory program leading to increased IL6 production.
Ischemic postconditioning (IPO) attenuates hepatic ischemia/reperfusion (I/R) injury. However, little is known about the underlying biological pathophysiology, which could be, at least in part, informed by exploring the transcriptomic changes using next-generation RNA sequencing (RNA-Seq). In this study, 18 mice (C57BL/6) were involved and randomly assigned to three groups: normal (n=6), I/R (n=6, subjected to 70% hepatic I/R), and IR+IPO (n=6, applying IPO to mice with I/R injury). We randomly selected 3 mice per group and extracted their liver tissues for next-generation RNA-Seq. We performed a bioinformatics analysis for two comparisons: normal vs. I/R and I/R vs. IR+IPO. From the analysis, 2416 differentially expressed genes (DEGs) were identified (p &lt; 0.05 and fold change ≥ 1.5). Gene ontology (GO) analysis revealed that these genes were mainly related to cellular metabolic processes, nucleic acids and protein binding processes. The enriched Kyoto Encyclopedia of Genes and Genomes (KEGG) pathways for the DEGs were the mitogen-activated protein kinase (MAPK), IL-17 signalling pathway, regulating pluripotency of stem cells, and insulin resistance pathway. Validation of 12 selected DEGs by qRT-PCR showed that Cyr61, Atf3, Nr4a1, Gdf15, Osgin1, Egr1, Epha2, Dusp1, Dusp6, Gadd45a and Gadd45b were significantly amplified. Finally, a protein-protein interaction (PPI) network constructed to determine interactions of these 11 DEGs. In summary, by exploring gene expression profiling in regard to hepatic I/R and IPO using next-generation RNA-Seq, we suggested a few progression-related genes and pathways, providing some clues for future experimental research.
Mitochondria are dynamic and plastic organelles, which flexibly adapt morphology, ATP production, and metabolic function to meet extrinsic challenges and demands. Regulation of mitochondrial biogenesis is essential during development and in adult life to survive stress and pathological insults, and is achieved not only by increasing mitochondrial mass, but also by remodeling the organellar proteome, metabolome, and lipidome. In the last decade, the post-transcriptional regulation of the expression of nuclear-encoded mitochondrial proteins has emerged as a fast, flexible, and powerful mechanism to shape mitochondrial function and coordinate it with other cellular processes. At the heart of post-transcriptional responses are a number of RNA-binding proteins that specifically bind mRNAs encoding mitochondrial proteins and define their fate, by influencing transcript maturation, stability, translation, and localization. Thus, RNA-binding proteins provide a uniquely complex regulatory code that orchestrates mitochondrial function during physiological and pathological conditions.
The sequences and structures of 3'-untranslated regions (3'UTRs) of messenger RNAs govern their stability, localization, and expression. 3'UTR regulatory elements are recognized by a wide variety of trans-acting factors that include microRNAs (miRNAs), their associated machinery, and RNA-binding proteins (RBPs). In turn, these factors instigate common mechanistic strategies to execute the regulatory programs encoded by 3'UTRs. Here, we review classes of factors that recognize 3'UTR regulatory elements and the effector machineries they guide toward mRNAs to dictate their expression and fate. We outline illustrative examples of competitive, cooperative, and coordinated interplay such as mRNA localization and localized translation. We further review the recent advances in the study of mRNP granules and phase transition, and their possible significance for the functions of 3'UTRs. Finally, we highlight some of the most recent strategies aimed at deciphering the complexity of the regulatory codes of 3'UTRs, and identify some of the important remaining challenges.
The Ca2+-binding protein Kv channel interacting protein 3 (KChIP3) or downstream regulatory element antagonist modulator (DREAM), a member of the neuronal calcium sensor (NCS) family, shows remarkable multifunctional properties. It acts as a transcriptional repressor in the nucleus and a modulator of ion channels or receptors, such as Kv4, NMDA receptors and TRPV1 channels on the cytomembrane. Previous studies of Kcnip3-/- mice have indicated that KChIP3 facilitates pain hypersensitivity by repressing Pdyn expression in the spinal cord. Conversely, studies from transgenic daDREAM (dominant active DREAM) mice indicated that KChIP3 contributes to analgesia by repressing Bdnf expression and attenuating the development of central sensitization. To further determine the role of KChIP3 in pain transmission and its possible involvement in emotional processing, we assessed the pain sensitivity and negative emotional behaviors of Kcnip3-/- rats. The knockout rats showed higher pain sensitivity compared to the wild-type rats both in the acute nociceptive pain model and in the late phase (i.e., 2, 4 and 6 days post complete Freund's adjuvant injection) of the chronic inflammatory pain model. Importantly, Kcnip3-/- rats displayed stronger aversion to the pain-associated compartment, higher anxiety level and aggravated depression-like behavior. Furthermore, RNA-Seq transcriptional profiling of the forebrain cortex were compared between wild-type and Kcnip3-/- rats. Among the 68 upregulated genes, 19 genes (including Nr4a2, Ret, Cplx3, Rgs9, and Itgad) are associated with neural development or synaptic transmission, particularly dopamine neurotransmission. Among the 79 downregulated genes, 16 genes (including Col3a1, Itm2a, Pcdhb3, Pcdhb22, Pcdhb20, Ddc, and Sncaip) are associated with neural development or dopaminergic transmission. Transcriptional upregulation of Nr4a2, Ret, Cplx3 and Rgs9, and downregulation of Col3a1, Itm2a, Pcdhb3 and Ddc, were validated by qPCR analysis. In summary, our studies showed that Kcnip3-/- rats displayed higher pain sensitivity and stronger negative emotions, suggesting an involvement of KChIP3 in negative emotions and possible role in central nociceptive processing.
Bovine leukemia virus (BLV) is a deltaretrovirus that infects domestic cattle. The structural protein Gag, found in all retroviruses, is a polyprotein comprising three major functional domains: matrix (MA), capsid (CA), and nucleocapsid (NC). Previous studies have shown that both mature BLV MA and NC are able to bind to nucleic acids; however, the viral assembly process and packaging of viral genomic RNA requires full-length Gag to produce infectious particles. Compared to lentiviruses, little is known about the structure of the Gag polyprotein of deltaretroviruses. In this work, structural models of full-length BLV Gag and Gag lacking the MA domain were generated based on previous structural data of individual domains, homology modeling, and flexible fitting to SAXS data using molecular dynamics. The models were used in molecular dynamic simulations to determine the relative mobility of the protein backbone. Functional annealing assays revealed the role of MA in the nucleic acid chaperone activity of BLV Gag. Our results show that full-length BLV Gag has an elongated rod-shaped structure that is relatively rigid, with the exception of the linker between the MA and CA domains. Deletion of the MA domain maintains the elongated structure but alters the rate of BLV Gag-facilitated annealing of two complementary nucleic acids. These data are consistent with a role for the MA domain of retroviral Gag proteins in modulating nucleic acid binding and chaperone activity. IMPORTANCE: BLV is a retrovirus that is found worldwide in domestic cattle. Since BLV infection has serious implications for agriculture, and given its similarities to human retroviruses such as HTLV-1, the development of an effective treatment would have numerous benefits. The Gag polyprotein exists in all retroviruses and is a key player in viral assembly. However, the full-length structure of Gag from any virus has yet to be elucidated at high resolution. This study provides structural data for BLV Gag and could be a starting point for modeling Gag-small molecule interactions with the ultimate goal of developing of a new class of pharmaceuticals.
Trypanosoma cruzi, the etiological agent of Chagas' disease, affects 8 million people predominantly living in socioeconomic underdeveloped areas. T. cruzi trypomastigotes (Ty), the classical infective stage, interact with the extracellular matrix (ECM), an obligatory step before invasion of almost all mammalian cells in different tissues. Here we have characterized the proteome and phosphoproteome of T. cruzi trypomastigotes upon interaction with ECM (MTy) and the data are available via ProteomeXchange with identifier PXD010970. Proteins involved with metabolic processes (such as the glycolytic pathway), kinases, flagellum and microtubule related proteins, transport-associated proteins and RNA/DNA binding elements are highly represented in the pool of proteins modified by phosphorylation. Further, important metabolic switches triggered by this interaction with ECM were indicated by decreases in the phosphorylation of hexokinase, phosphofructokinase, fructose-2,6-bisphosphatase, phosphoglucomutase, phosphoglycerate kinase in MTy. Concomitantly, a decrease in the pyruvate and lactate and an increase of glucose and succinate contents were detected by GC-MS. These observations led us to focus on the changes in the glycolytic pathway upon binding of the parasite to the ECM. Inhibition of hexokinase, pyruvate kinase and lactate dehydrogenase activities in MTy were observed and this correlated with the phosphorylation levels of the respective enzymes. Putative kinases involved in protein phosphorylation altered upon parasite incubation with ECM were suggested by in silico analysis. Taken together, our results show that in addition to cytoskeletal changes and protease activation, a reprogramming of the trypomastigote metabolism is triggered by the interaction of the parasite with the ECM prior to cell invasion and differentiation into amastigotes, the multiplicative intracellular stage of T. cruzi in the vertebrate host.
Local translation in neurites is a phenomenon that enhances the spatial segregation of proteins and their functions away from the cell body, yet it is unclear how local translation varies across neuronal cell types. Further, it is unclear whether differences in local translation across cell types simply reflect differences in transcription or whether there is also a cell type-specific post-transcriptional regulation of the location and translation of specific mRNAs. Most of the mRNAs discovered as being locally translated have been identified from hippocampal neurons because their laminar organization facilitates neurite-specific dissection and microscopy methods. Given the diversity of neurons across the brain, studies have not yet analyzed how locally translated mRNAs differ across cell types. Here, we used the SynapTRAP method to harvest two broad cell types in the mouse forebrain: GABAergic neurons and layer 5 projection neurons. While some transcripts overlap, the majority of the local translatome is not shared across these cell types. In addition to differences driven by baseline expression levels, some transcripts also exhibit cell type-specific post-transcriptional regulation. Finally, we provide evidence that GABAergic neurons specifically localize mRNAs for peptide neurotransmitters, including somatostatin and cortistatin, suggesting localized production of these key signaling molecules in the neurites of GABAergic neurons. Overall, this work suggests that differences in local translation in neurites across neuronal cell types are poised to contribute substantially to the heterogeneity in neuronal phenotypes.
N-Methylation of DNA/RNA bases can be regulatory or damaging and is linked to diseases including cancer and genetic disorders. Bacterial AlkB and human FTO are DNA/RNA demethylases belonging to the Fe(ii) and 2-oxoglutarate oxygenase superfamily. Modelling studies reveal conformational dynamics influence structure-function relationships of AlkB and FTO, e.g. why 1-methyladenine is a better substrate for AlkB than 6-methyladenine. Simulations show that the flexibility of the double stranded DNA substrate in AlkB influences correlated motions, including between the core jelly-roll fold and an active site loop involved in substrate binding. The FTO N- and C-terminal domains move in respect to one another in a manner likely important for substrate binding. Substitutions, including clinically observed ones, influencing catalysis contribute to the network of correlated motions in AlkB and FTO. Overall, the calculations highlight the importance of the overall protein environment and its flexibility to the geometry of the reactant complexes.
An oncolytic herpes simplex virus (oHSV) has proven amenable in oncolytic virotherapy and was approved to treat melanoma. The immediate-early (IE) protein ICP27 encoded by gene UL54 is essential for HSV infection. Post-transcriptional modification of UL54 would increase tumor targeting of oHSVs. However, UL54 gene transcription regulatory sequences and factors were not reported yet. Here we isolated a new strain LXMW of type 1 HSV (HSV-1-LXMW) in China and found it's closely related to HSV-1 strains Patton and H129 in the US by the first and next generation DNA sequencing viral DNA phylogenetic analysis. Using a weight matrix-based program Match, we found the UL54 transcription regulatory sequences binding to the transcription factors Oct-1, v-Myb and Pax-6 in HSV-1-LXMW, while the sequences binding to Oct-1 and Hairy in a HSV-2 strain. Further validation showed that HSV-1 and HSV-2 shared the common sequence binding to Oct-1, but had unique sequences to bind v-Myb and Pax-6, or Hairy, respectively, by DNA sequence alignment of total 11 HSV strains. The published results howed that the expression of transcription factors is consistent with the tissue tropism of HSV-1 and HSV-2. In the current article a new HSV-1 strain LXMW was isolated and its putative HSV UL54 transcription regulatory sequences and factors were identified for the first time. Our findings highlight the new understanding of the principles of transcriptional regulation in HSV biology and oncolytic virotherapy.
The original version of this article contained error in Figure 2e. In Figure 2e, the 6th colony image of T47D cells treated with shMSI2 was inadvertently replaced with a duplicate of 7th colony image. However, the conclusions reported in the manuscript are not affected by figure replacement. The authors regret that these errors were made and apologize for the confusion and inconvenience. The correct version of this figure panel appears in the Author Correction associated with this Article.
We have developed a homogeneous time-resolved fluorescence (HTRF)-based enzyme assay to measure the catalytic activity of N6-methyladenosine (m6A) methyltransferases and demethylases. The assay detects m6A modifications using the natural m6A-binding proteins (m6A readers). The reaction product or substrate m6A-containing RNA and the m6A reader protein are fluorescently labeled such that their proximity during binding initiates Förster resonance energy transfer (FRET). We show that our HTRF assay can be used for high-throughput screening, which will facilitate the discovery of small-molecule modulators of m6A (de)methylases.
UV crosslinking and immunoprecipitation (CLIP) coupled with high-throughput sequencing is the most state-of-the-art technology to characterize protein-RNA interactions, yet only a small portion of RNA-binding proteins (RBPs) have been studied by CLIP due to its complex procedures and high level of false-positive signals. Herein, we report a SpyCLIP method that employs a covalent linkage formed between the RBP-fused SpyTag and SpyCatcher, which can withstand the harshest washing conditions for removing nonspecific interactions. Moreover, SpyCLIP circumvents the radioactive labeling and PAGE-membrane purification steps, and the whole procedure can be performed on beads and is readily amenable to automation. We investigated multiple RBPs by SpyCLIP and generated high-quality RNA binding maps with significantly improved reproductivity and accuracy. Therefore, the small tag size and convenient protocol of SpyCLIP provides a robust method for both routine characterization and high-throughput studies of protein-RNA interactions.
Glycogen storage disease type-Ia (GSD-Ia), caused by a deficiency in glucose-6-phosphatase-α (G6Pase-α or G6PC), is characterized by impaired glucose homeostasis with a hallmark hypoglycemia, following a short fast. We have shown that G6pc-deficient (G6pc-/-) mice treated with recombinant adeno-associated virus (rAAV) vectors expressing either wild-type (WT) (rAAV-hG6PC-WT) or codon-optimized (co) (rAAV-co-hG6PC) human (h) G6Pase-α maintain glucose homeostasis if they restore ≥3% of normal hepatic G6Pase-α activity. The co vector, which has a higher potency, is currently being used in a phase I/II clinical trial for human GSD-Ia (NCT03517085). While routinely used in clinical therapies, co vectors may not always be optimal. Codon-optimization can impact RNA secondary structure, change RNA/DNA protein-binding sites, affect protein conformation and function, and alter posttranscriptional modifications that may reduce potency or efficacy. We therefore sought to develop alternative approaches to increase the potency of the G6PC gene transfer vectors. Using an evolutionary sequence analysis, we identified a Ser-298 to Cys-298 substitution naturally found in canine, mouse, rat, and several primate G6Pase-α isozymes, that when incorporated into the WT hG6Pase-α sequence, markedly enhanced enzymatic activity. Using G6pc-/- mice, we show that the efficacy of the rAAV-hG6PC-S298C vector was 3-fold higher than that of the rAAV-hG6PC-WT vector. The rAAV-hG6PC-S298C vector with increased efficacy, that minimizes the potential problems associated with codon-optimization, offers a valuable vector for clinical translation in human GSD-Ia.
The neurotoxin 6-hydroxydopamine (6-OHDA) has been widely exploited as a tool for modeling Parkinson's disease (PD) in the rat. This study aimed to provide a comprehensive profile of the mRNAs and long noncoding RNAs (lncRNAs) in rats treated with 6-OHDA as a model of PD. Female SPF Wistar rats were randomly divided into two groups: a PD model group and a control group. The PD model was induced by 6-OHDA injection. RNA-seq analysis was performed on 6-OHDA-treated rats and corresponding controls. Novel lncRNAs were identified. Differentially expressed genes (DEGs) and differentially expressed lncRNAs were identified in the PD group compared with controls. Gene Ontology function and pathway enrichment analyses were conducted on the DEGs, followed by construction of a protein-protein interaction (PPI) network. In addition, prediction of lncRNA target genes and function prediction of lncRNAs were performed. Moreover, microRNAs (miRNAs) that interacted with the DEGs and differentially expressed lncRNAs were predicted to construct a miRNA-lncRNA-mRNA regulatory network. A total of 536 DEGs and 512 differentially expressed lncRNAs (44 up-regulated and 10 down-regulated known lncRNAs; 407 up-regulated and 51 down-regulated novel lncRNAs) were identified in the PD rat model compared with controls. The DEGs and target genes of lncRNAs were mainly associated with the innate immune response, 2'-5'-oligoadenylate synthetase activity, GTPase activity, GTP binding and the RIG-I-like receptor signaling pathway. IRF7 and ISG15 were hub proteins in the PPI network. Many mRNAs and lncRNAs interacted with other molecules in a competing endogenous RNA network, such as MAS1, TMPRSS2, NPTX1, XLOC_016191, XLOC_026924 and XLOC_005439. We conclude that IRF7, ISG15, MAS1, TMPRSS2, NPTX1, XLOC_016191, XLOC_026924 and XLOC_005439 may contribute critical roles in the pathogenesis of PD.
Caveolins (CAVs) are structural proteins of caveolae that function as signaling platforms to regulate smooth muscle contraction. Loss of CAV protein expression is associated with impaired contraction in obstruction-induced bladder smooth muscle (BSM) hypertrophy. In this study, microarray analysis of bladder RNA revealed down-regulation of CAV1, CAV2, and CAV3 gene transcription in BSM from models of obstructive bladder disease in mice and humans. We identified and characterized regulatory regions responsible for CAV1, CAV2, and CAV3 gene expression in mice with obstruction-induced BSM hypertrophy, and in men with benign prostatic hyperplasia. DNA affinity chromatography and chromatin immunoprecipitation assays revealed a greater increase in binding of GATA-binding factor 6 (GATA-6) and NF-κB to their cognate binding motifs on CAV1, CAV2, and CAV3 promoters in obstructed BSM relative to that observed in control BSM. Knockout of NF-κB subunits, shRNA-mediated knockdown of GATA-6, or pharmacologic inhibition of GATA-6 and NF-κB in BSM increased CAV1, CAV2, and CAV3 transcription and promoter activity. Conversely, overexpression of GATA-6 decreased CAV2 and CAV3 transcription and promoter activity. Collectively, these data provide new insight into the mechanisms by which CAV gene expression is repressed in hypertrophied BSM in obstructive bladder disease.
Retinoids, such as all- trans-retinoic acid (ATRA), are endogenous signaling molecules derived from vitamin A that influence a variety of cellular processes through mediation of transcription events in the cell nucleus. Because of these wide-ranging and powerful biological activities, retinoids have emerged as therapeutic candidates of enormous potential. However, their use has been limited, to date, due to a lack of understanding of the complex and intricate signaling pathways that they control. We have designed and synthesized a family of synthetic retinoids that exhibit strong, intrinsic, solvatochromatic fluorescence as multifunctional tools to interrogate these important biological activities. We utilized the unique photophysical characteristics of these fluorescent retinoids to develop a novel in vitro fluorometric binding assay to characterize and quantify their binding to their cellular targets, including cellular retinoid binding protein II (CRABPII). The dihydroquinoline retinoid, DC360, exhibited particularly strong binding ( Kd = 34.0 ± 2.5 nM), and we further used X-ray crystallography to determine the structure of the DC360-CRABPII complex to 1.8 Å, which showed that DC360 occupies the known hydrophobic retinoid binding pocket. Finally, we used confocal fluorescence microscopy to image the cellular behavior of the compounds in cultured human epithelial cells, highlighting a fascinating nuclear localization, and used RNA sequencing to confirm that the compounds regulate cellular processes similar to those of ATRA. We anticipate that the unique properties of these fluorescent retinoids can now be used to cast new light on the vital and highly complex retinoid signaling pathway.
Endothelial barrier function depends on the proper localization and function of the adherens junction protein VE (vascular endothelial)-cadherin. Previous studies have suggested a functional relationship between integrin-mediated adhesion complexes and VE-cadherin yet the underlying molecular links are unclear. Binding of the cytoskeletal adaptor protein talin to the β-integrin cytoplasmic domain is a key final step in regulating the affinity of integrins for extracellular ligands (activation) but the role of integrin activation in VE-cadherin mediated endothelial barrier function is unknown.
To test the requirement of talin-dependent activation of β1 integrin in VE-cadherin organization and endothelial cell (EC) barrier function.
EC-specific deletion of talin in adult mice resulted in impaired stability of intestinal microvascular blood vessels, hemorrhage, and death. Talin-deficient endothelium showed altered VE-cadherin organization at EC junctions in vivo. shRNA (short hairpin RNA)-mediated knockdown of talin1 expression in cultured ECs led to increased radial actin stress fibers, increased adherens junction width and increased endothelial monolayer permeability measured by electrical cell-substrate impedance sensing. Restoring β1-integrin activation in talin-deficient cells with a β1-integrin activating antibody normalized both VE-cadherin organization and EC barrier function. In addition, VE-cadherin organization was normalized by reexpression of talin or integrin activating talin head domain but not a talin head domain mutant that is selectively deficient in activating integrins.
Talin-dependent activation of EC β1-integrin stabilizes VE-cadherin at endothelial junctions and promotes endothelial barrier function.
In this present study novel endoxylanase producing Bacillus licheniformis DM5 isolated, identified based on 16S rDNA from Garampani hotspring, Assam, India and enzyme was purified. RNA secondary structure predicted the similarity of B. licheniformis DM5 with B. licheniformis ATCC14580. Highest production of xylanase from B. licheniformis DM5 was achieved in the TY medium with cell densities 12 g/l and extracellular protein concentration containing xylanase 400 mg/l. Partially purified extracellular xylanase displayed optimum pH 6.5 and temperature 50 °C. Thermostability of the xylanase at the elevated temperature showed stability between 50 and 60 °C retaining its 99% activity. Kinetic parameters of thermophilic xylanase revealed Km 1.5 ± 0.2 mg/ml, Vmax 2.7 ± 0.2 U/ml and and Kcat 1.8 ± 0.2 s-1 against beechwood xylan but ruled out any exo-acting activity against synthetic pNP-xylopyranoside substrate. Time dependent enzymatic hydrolysis of beechwood xylan and preprocessed agrowaste corncob exhibited the release of xylotriose and xylobiose oligosaccharide (XOS) significantly high. Xylobiose and xylotriose exhibited higher binding affinities with BIAXP transporter protein of probiotic bacteria explaining their easy uptake by the cells. Mixed oligosaccharides also exhibited better prebiotic activity by promoting growth of Bifidobacterium infantis and Lactobacillus delbrueckii. Mixed XOS when tested for their cytotoxicity on Hela cell lines in in vitro MTT assay displayed significant lowering of cell viability after 48 h and 24 h at 100 µg/ml to 60% and 50%, respectively. In contrast, cytotoxicity wasn't observed against normal cervical cell line (VK2/E6E7-ATCC-CRL-2616). Therefore, thermophilic endoxylanase from B. licheniformis DM5 could be attributed for the production of prebiotic and anti-inflammatory XOS from agrowaste.
To figure out the relationship between SMAD3 and serine-threonine tyrosine kinase (STYK1) in ovarian carcinoma cell's paclitaxel resistance.
The quantitative reverse transcription-polymerase chain reactpostion and Western blot analysis were used to analyze RNA and protein content of SMAD3 and STYK1, respectively. The chromatin immunoprecipitation assay was used to confirm the binding site of SMAD3 to the STYK1 promoter region. Transwell assay was used to detect cell invasion and migration, and Western Blot was used to detect the marker proteins (vimentin and E-cadherin) of epithelial-mesenchymal transition (EMT) process. MTT and apoptosis assay were used to, respectively, measure cell vitality and apoptosis. In vivo experiments, rats were subcutaneously implanted with A2780 cells to establish an animal model of ovarian cancer and the survival curve was drawn.
Upregulating SMAD3 induced the expression of STYK1 in ovarian cancer cell lines. STYK1 is a direct transcriptional target of SMAD3. Upregulating STYK1 improved the paclitaxel resistance of ovarian carcinoma cells. Upregulating STYK1 promoted cell invasion, migration, and the EMT process, and SMAD3 had the same effect with STYK1 on cell invasion, cell migration, and the EMT process. The animal assay showed that downregulating STYK1 inhibited the EMT process and the paclitaxel resistance, further promoting the treatment of cervical cancer.
SMAD3 combined with the promoter region of STYK1 to promote the transcription process of STYK1, thereby promoting the EMT process and paclitaxel resistance of ovarian cancer cells.
Long non-coding RNAs (lncRNAs) can act as scaffolds that promote the interaction of proteins, RNA, and DNA. There is increasing evidence of sequence-specific interactions of lncRNAs with DNA via triple-helix (triplex) formation. This process allows lncRNAs to recruit protein complexes to specific genomic regions and regulate gene expression. Here we propose a computational method called Triplex Domain Finder (TDF) to detect triplexes and characterize DNA-binding domains and DNA targets statistically. Case studies showed that this approach can detect the known domains of lncRNAs Fendrr, HOTAIR and MEG3. Moreover, we validated a novel DNA-binding domain in MEG3 by a genome-wide sequencing method. We used TDF to perform a systematic analysis of the triplex-forming potential of lncRNAs relevant to human cardiac differentiation. We demonstrated that the lncRNA with the highest triplex-forming potential, GATA6-AS, forms triple helices in the promoter of genes relevant to cardiac development. Moreover, down-regulation of GATA6-AS impairs GATA6 expression and cardiac development. These data indicate the unique ability of our computational tool to identify novel triplex-forming lncRNAs and their target genes.
RNA-binding proteins (RBPs) control mRNA processing, stability, transport, editing, and translation. We recently conducted transcriptome analyses comparing normal (i.e., healthy) cervical tissue samples with human papillomavirus (HPV)-positive cervical cancer tissue samples and identified 614 differentially expressed protein-coding transcripts which are enriched in cancer-related pathways and consist of 95 known RBPs. We verified the altered expression of 26 genes with a cohort of 72 cervical samples, including 24 normal cervical samples, 25 cervical intraepithelial neoplasia grade 2 (CIN2) and CIN3 samples, and 23 cervical cancer tissue samples. LY6K (lymphocyte antigen 6 complex locus K), FAM83A (family member with sequence similarity 83), CELSR3, ASF1B, IQGAP3, SEMA3F, CLDN10, MSX1, CXCL5, ASRGL1, ELAVL2, GRB7, KHSRP, NOVA1, PTBP1, and RNASEH2A were identified as novel candidate genes associated with cervical lesion progression and carcinogenesis. HPV16 or HPV18 infection was found to alter the expression of 8 RBP genes (CDKN2A, ELAVL2, GRB7, HSPB1, KHSRP, NOVA1, PTBP1, and RNASEH2A) in human vaginal and foreskin keratinocytes. Both viral E6 and E7 decreased NOVA1 expression, but only E7 increased the expression of RNASEH2A in an E2F1-dependent manner. Proliferating cell nuclear antigen (PCNA) directs RNASEH2 activity with respect to DNA replication by removing the RNA primers to promote Okazaki fragment maturation, and two factors are closely associated with neoplasia progression. Therefore, we predict that the induction of expression of RNASEH2A via viral E7 and E2F1 may promote DNA replication and cancer cell proliferation.IMPORTANCE High-risk HPV infections lead to development of cervical cancer. This study identified the differential expression of 16 novel genes (LY6K, FAM83A, CELSR3, ASF1B, IQGAP3, SEMA3F, CLDN10, MSX1, CXCL5, ASRGL1, ELAVL2, GRB7, KHSRP, NOVA1, PTBP1, and RNASEH2A) in HPV-infected cervical tissue samples and keratinocytes. Eight of these genes (CDKN2A, ELAVL2, GRB7, HSPB1, KHSRP, NOVA1, PTBP1, and RNASEH2A) encode RNA-binding proteins. Further studies indicated that both HPV16 and HPV18 infections lead to the aberrant expression of selected RBP-encoding genes. We found that viral E6 and E7 decrease NOVA1 expression but that E7 increases RNASEH2A expression via E2F1. The altered expression of these genes may be utilized as biomarkers for high-risk (HR)-HPV carcinogenesis and progression.
Neonatal sepsis remains a leading cause of infant mortality. Cold-inducible RNA binding protein (CIRP) is an inflammatory mediator that induces TNF-α production in macrophages. C23 is a CIRP-derived peptide that blocks CIRP from binding its receptor. We therefore hypothesized that treatment with C23 reduces systemic inflammation and protects the lungs in neonatal sepsis.
Sepsis was induced in C56BL/6 mouse pups (5-7 days) by intraperitoneal injection of adult cecal slurry (0.525 mg/g body weight, LD100). One hour later pups received retroorbital injection of C23 (8 mg/kg) or vehicle (normal saline). Ten hours after sepsis induction, blood and tissues were collected for analysis.
C23 treatment resulted in a 58% and 69% reduction in serum levels of proinflammatory cytokines IL-6 and IL-1β, respectively, and a 40% and 45% reduction of AST and LDH, as compared to vehicle-treated septic pups. In the lungs, C23 treatment reduced expression of cytokines IL-6 and IL-1β by 78% and 74%. In addition, the mRNA level of neutrophil chemoattractants KC and MIP-2 was reduced by 84% and 74%, respectively. These results corresponded to a reduction in histologic lung injury score. Vehicle-treated pups scored 0.49 ± 0.19, while C23 treatment reduced scores to 0.29 ± 0.12 (p &lt; 0.05; Max = 1). Apoptosis in the lungs, measured by TUNEL assay, was also decreased by 53% with C23 treatment (p &lt; 0.05).
Inhibition of CIRP with C23 treatment is protective in septic neonatal mice as demonstrated by reduced inflammatory markers systemically and in the lung. Therefore, C23 has promising therapeutic potential in treatment of neonatal sepsis.
Level I.
DNA hydroxymethylation has gained attention as an intermediate in the process of DNA demethylation. More recently, 5-hydroxymethylcytosine has been recognized as an independent epigenetic mark that can persist over time and that exerts influence on gene regulation and other biological processes. Deregulation of this DNA modification has been linked to tumorigenesis and a variety of other diseases. The impact of irradiation on DNA hydroxymethylation is poorly understood. In this study we exposed lung fibroblasts (IMR90) to 0.5 Gy and 2 Gy of X-rays, respectively. We characterized radiation induced changes of DNA hydroxymethylation 1 h, 6 h, 24 h and 120 h after exposure employing immunoprecipitation and subsequent deep sequencing of the genomic fraction enriched for hydroxymethylated DNA. Transcriptomic response to irradiation was analyzed for time points 6 h and 24 h post exposure by means of RNA sequencing. Irradiated and sham-irradiated samples shared the same overall distribution of 5-hydroxymethylcytosines with respect to genomic features such as promoters and exons. The frequency of 5-hydroxymethylcytosine peaks differentially detected in irradiated samples increased in genic regions over time, while the opposing trend was observed for intergenic regions. Onset and extent of this effect was dose dependent. Moreover, we demonstrated a biased distribution of 5-hmC alterations at CpG islands and sites occupied by the DNA binding protein CTCF. In summary, our study provides new insights into the epigenetic response to irradiation. Our data highlight genomic features more prone to irradiation induced changes of DNA hydroxymethylation, which might impact early and late onset effects of irradiation.
Saturated fatty acid (SFA)-induced lipotoxicity is caused by the accumulation of reactive oxygen species (ROS), which is associated with damaged mitochondria. Moreover, lipotoxicity is crucial for the progression of nonalcoholic steatohepatitis (NASH). Autophagy is required for the clearance of protein aggregates or damaged mitochondria to maintain cellular metabolic homeostasis. The NFE2L2/NRF2 (nuclear factor, erythroid 2 like 2)-KEAP1 (kelch like ECH associated protein 1) pathway is essential for the elimination of ROS. ULK1 (unc-51 like autophagy activating kinase 1; yeast Atg1) is involved in the initiation of autophagy; however, its role in lipotoxicity-induced cell death in hepatocytes and mouse liver has not been elucidated. We now show that ULK1 potentiates the interaction between KEAP1 and the autophagy adaptor protein SQSTM1/p62, thereby mediating NFE2L2 activation in a manner requiring SQSTM1-dependent autophagic KEAP1 degradation. Furthermore, ULK1 is required for the autophagic removal of damaged mitochondria and to enhance binding between SQSTM1 and PINK1 (PTEN induced kinase 1). This study demonstrates the molecular mechanisms underlying the cytoprotective role of ULK1 against lipotoxicity. Thus, ULK1 could represent a potential therapeutic target for the treatment of NASH. Abbreviations: ACTB: actin beta; CM-H2DCFDA:5-(and-6)-chloromethyl-2',7'-dichlorodihydrofluorescein diacetate; CQ: chloroquine; CUL3: cullin 3; DMSO: dimethyl sulfoxide; GSTA1: glutathione S-transferase A1; HA: hemagglutinin; Hepa1c1c7: mouse hepatoma cells; HMOX1/HO-1: heme oxygenase 1; KEAP1: kelch like ECH associated protein 1; LPS: lipopolysaccharides; MAP1LC3/LC3: microtubule-associated protein 1 light chain 3; MAPK8/JNK: mitogen-activated protein kinase 8; MEF: mouse embryonic fibroblast; MFN1: mitofusin 1; MTT: 3-(4,5-dimethylthiazol-2-yl)-2,5-diphenyltetrazolium bromide; NASH: nonalcoholic steatohepatitis; NFE2L2/NRF2: nuclear factor, erythroid 2 like 2; NQO1: NAD(P)H quinone dehydrogenase 1; PA: palmitic acid; PARP: poly (ADP-ribose) polymerase 1; PINK1: PTEN induced kinase 1; PRKAA1/2: protein kinase AMP-activated catalytic subunits alpha1/2; PRKN/PARK2: parkin RBR E3 ubiquitin protein ligase; PRKC/PKC: protein kinase C; RBX1: ring-box 1; ROS: reactive oxygen species; SFA: saturated fatty acid; siRNA: small interfering RNA; SQSTM1/p62: sequestosome 1; TOMM20: translocase of outer mitochondrial membrane 20; TUBA: tubulin alpha; TUNEL: terminal deoxynucleotidyl transferase-mediated dUTP nick-end labeling; ULK1: unc-51 like autophagy activating kinase 1.
Accumulating evidence indicates that long non-coding RNAs (lncRNAs) can play a pivotal role in regulation of diverse cellular processes. In particular, lncRNAs can serve as master gene regulators at transcriptional and posttranscriptional levels, leading to tumorigenesis. In this review, we discuss latest developments in lncRNA-meditated gene expression at the post-transcriptional level, including gene splicing, mRNA stability, protein stability and nuclear trafficking.
Ulcerative colitis (UC) is an important risk factor for the occurrence of colon cancer, and changes in expression of p53 and inflammatory factors are closely related to the pathogenesis of colon cancer. Therefore, changes in expression of p53 and inflammatory factors in UC were investigated to explore its intrinsic pathogenetic laws. The levels of inflammatory factors, such as interleukin-6 (IL-6), tumor necrosis factor-α (TNF-α), IL-10 and IL-4, in peripheral blood of UC patients and healthy adults were detected via enzyme-linked immunosorbent assay, and the changes in p53 expression were analyzed via immunohistochemistry and western blotting, and the correlation of p53 expression with changes in cytokines was also analyzed. Moreover, changes in 45S preribosomal ribonucleic acid (preRNA) transcriptional activity in four kinds of cell lines exposed to IL-6 were analyzed and determined by using reverse transcription-quantitative polymerase chain reaction. Finally, the C-myc protein expression after IL-6 stimulation was analyzed and evaluated via western blot analysis. The levels of IL-6 and TNF-α in peripheral blood in the UC patient group were significantly increased compared with those in the healthy adult group (P&lt;0.01), while the levels of IL-10 and IL-4 in peripheral blood were significantly decreased compared with those in the healthy adult group (P&lt;0.01). The p53 expression had a significant negative correlation with IL-6. The results showed that IL-6 activated C-myc messenger RNA (mRNA) translation, thereby upregulating the ribosomal RNA (rRNA) transcription. Additionally, IL-6 stimulated the mouse double minute 2 homolog (MDM2)-mediated proteasomal degradation of p53 by reducing the availability of ribosomal protein used for MDM2 binding, thus resulting in the downregulation of p53 protein expression. The findings of the study show that, expression level of IL-6 was increased in UC patients, which regulates the downregulation of p53 expression level and plays a role in tumorigenesis through enhancing cell proliferation and inhibiting apoptosis.
Hot spot residues at protein⁻RNA complexes are vitally important for investigating the underlying molecular recognition mechanism. Accurately identifying protein⁻RNA binding hot spots is critical for drug designing and protein engineering. Although some progress has been made by utilizing various available features and a series of machine learning approaches, these methods are still in the infant stage. In this paper, we present a new computational method named XGBPRH, which is based on an eXtreme Gradient Boosting (XGBoost) algorithm and can effectively predict hot spot residues in protein⁻RNA interfaces utilizing an optimal set of properties. Firstly, we download 47 protein⁻RNA complexes and calculate a total of 156 sequence, structure, exposure, and network features. Next, we adopt a two-step feature selection algorithm to extract a combination of 6 optimal features from the combination of these 156 features. Compared with the state-of-the-art approaches, XGBPRH achieves better performances with an area under the ROC curve (AUC) score of 0.817 and an F1-score of 0.802 on the independent test set. Meanwhile, we also apply XGBPRH to two case studies. The results demonstrate that the method can effectively identify novel energy hotspots.
Small non-coding RNAs control normal development and differentiation in the embryo. These regulatory molecules play a key role in the development of human diseases and are used often today for researching new treatments for different pathologies. In this study, CaCo2 colorectal adenocarcinoma cells were initially epigenetically reprogrammed and transformed into CD4+ cells with nano-sized complexes of amphiphilic poly-(N-vinylpyrrolidone) (PVP) with miRNA-152 and piRNA-30074. The transformation of cells was confirmed by morphological and genetic changes in the dynamic of reprogramming. CD4+ lymphocytes marker was detected using immunofluorescence. Amphiphilic poly-(N-vinylpyrrolidone)/small non-coding RNAs complexes were investigated for transfection efficiency and duration of transfection of CaCo2 colorectal adenocarcinoma cells using fluorescence.
Since the identification of the first RNA demethylase and the establishment of methylated RNA immunoprecipitation-sequencing methodology 6 to 7 years ago, RNA methylation has emerged as a widespread phenomenon and a critical regulator of transcript expression. This new layer of regulation is termed "epitranscriptomics." The most prevalent RNA methylation, N6-methyladenosine (m6A), occurs in approximately 25% of transcripts at the genome-wide level and is enriched around stop codons, in 5'- and 3'-untranslated regions, and within long internal exons. RNA m6A modification regulates RNA splicing, translocation, stability, and translation into protein. m6A is catalyzed by the RNA methyltransferases METTL3, METTL14, and METTL16 (writers), is removed by the demethylases FTO and ALKBH5 (erasers), and interacts with m6A-binding proteins, such as YTHDF1 and IGF2BP1 (readers). RNA methyltransferases, demethylases, and m6A-binding proteins are frequently upregulated in human cancer tissues from a variety of organ origins, increasing onco-transcript and oncoprotein expression, cancer cell proliferation, survival, tumor initiation, progression, and metastasis. Although RNA methyltransferase inhibitors are not available yet, FTO inhibitors have shown promising anticancer effects in vitro and in animal models of cancer. Further screening for selective and potent RNA methyltransferase, demethylase, or m6A-binding protein inhibitors may lead to compounds suitable for future clinical trials in cancer patients.
MazF is a sequence-specific endoribonuclease that is widely conserved in bacteria and archaea. Here, we found an MazF homologue (MazF-lp; LPO-p0114) in Legionella pneumophila. The mazF-lp gene overlaps 14 base pairs with the upstream gene mazE-lp (MazE-lp; LPO-p0115). The induction of mazF-lp caused cell growth arrest, while mazE-lp co-induction recovered cell growth in Escherichia coli. In vivo and in vitro primer extension experiments showed that MazF-lp is a sequence-specific endoribonuclease cleaving RNA at AACU. The endoribonuclease activity of purified MazF-lp was inhibited by purified MazE-lp. We found that MazE-lp and the MazEF-lp complex specifically bind to the palindromic sequence present in the 5'-untranslated region of the mazEF-lp operon. MazE-lp and MazEF-lp both likely function as a repressor for the mazEF-lp operon and for other genes, including icmR, whose gene product functions as a secretion chaperone for the IcmQ pore-forming protein, by specifically binding to the palindromic sequence in 5'-UTR of these genes.
RNA structure is intimately connected to each step of gene expression. Recent advances have enabled transcriptome-wide maps of RNA secondary structure, called 'RNA structuromes'. However, previous whole-cell analyses lacked the resolution to unravel the landscape and also the regulatory mechanisms of RNA structural changes across subcellular compartments. Here we reveal the RNA structuromes in three compartments, chromatin, nucleoplasm and cytoplasm, in human and mouse cells. The cytotopic structuromes substantially expand RNA structural information and enable detailed investigation of the central role of RNA structure in linking transcription, translation and RNA decay. We develop a resource with which to visualize the interplay of RNA-protein interactions, RNA modifications and RNA structure and predict both direct and indirect reader proteins of RNA modifications. We also validate a novel role for the RNA-binding protein LIN28A as an N6-methyladenosine modification 'anti-reader'. Our results highlight the dynamic nature of RNA structures and its functional importance in gene regulation.
The aim of this study was to evaluate expression of Post-Acrosomal WW Binding Protein (PAWP) in infertile men with low and high fertilization post ICSI and also globozoospermic men.
Semen samples were collected from 18 infertile men with previous failed or low fertilization (&lt;25%) post ICSI, 10 men with high fertilization (&gt;50%) post ICSI, 15 globozoospermic men, and 21 fertile individuals. Then, expression of PAWP was assessed at RNA with quantitative Real Time PCR.
Relative expression of PAWP in sperm was significantly (p&lt;0.05) lower in infertile men with globozoospermia (41.5±5.7) or low fertilization rate (43.3±10.4) compared to fertile (138.8±17.3)or men with high fertilization (211.6±75.6). In addition, a significant positive correlation (r=0.628; p=0.001) was observed between percentage of fertilization with the relative expression of PAWP.
Considering solid recent evidences regarding PLC? as the main sperm factor involved in oocyte activation, therefore co-localization of PLC? with PAWP in perinuclear theca may account for the above observation and it is likely that PAWP may have other functions and/or it may assist PLC?.
A better understanding of the underlying molecular mechanisms in treatment failure of bevacizumab (BEV) for malignant glioma would contribute to overcome therapeutic resistance.
Here, we used a quantitative proteomic method to identify molecular signatures of glioblastoma cell after BEV treatment by two-dimensional liquid chromatography-tandem mass spectrometry analysis and 6-plex iTRAQ quantification. Next, the function of cold-inducible RNA-binding protein (CIRP), one of the most significantly affected proteins by drug treatment, was evaluated in drug resistance of glioma cells by invasion assays and animal xenograft assays. Target molecules bound by CIRP were determined using RNA-binding protein immunoprecipitation and microarray analysis. Then, these mRNAs were identified by quantitative real-time PCR.
Eighty-seven proteins were identified with significant fold changes. The biological functional analysis indicated that most of the proteins were involved in the process of cellular signal transduction, cell adhesion, and protein transport. The expression of CIRP greatly decreased after BEV treatment, and ectopic expression of CIRP abolished cell migration in BEV-treated glioma cells. In addition, CIRP could bind mRNA of CXCL12 and inhibit BEV-induced increase of CXCL12 in glioma cells.
These data suggested that CIRP may take part in BEV-induced migration of gliomas by binding of migration-relative RNAs.
Rotavirus is the representative cause of severe acute gastroenteritis in young children. A characteristic feature of rotavirus is low infectious dose and robustness of the virion, suggesting sanitation and hygiene will have little impact. Thus, development of a vaccine should be given priority. Efficient capture of infectious viruses is an important step in generating a vaccine. Previously, antibody-integrated magnetic nanobeads (MNBs) have been developed for virus capture. This study examines the applicability of this method for infectious rotavirus recovery and enrichment.
Graphite-encapsulated MNBs were treated with radio frequency (RF) excited Ar/NH3 gas mixture plasma to introduce amino groups onto their surfaces. Rotavirus viral protein 7 (VP7) antibody was attached to the surface of MNBs via these amino groups using a coupling agent, N-succinimidyl 3-(2-pyridyldithio)propionate (SPDP). The antibody-integrated MNBs were incubated with rotavirus-infected cell lysate and then separated from the supernatant by applying a magnetic field. After thorough washing, rotavirus was recovered and enrichment analysis done by polymerase chain reaction (PCR), immunochromatography, and an infection analysis using MA104 cells.
Immunochromatography and PCR indicate that anti-rotavirus antibody-integrated MNPs efficiently capture rotavirus with the capsid protein and viral RNA. The estimated recovery rate was 80.2% by PCR and 90.0% by infection analysis, while the concentrating factor was 7.9-fold by PCR and 6.7-fold by infection analysis. In addition, the absence of non-specific binding to the antibody-integrated MNPs was confirmed using anti-dengue virus antibody-integrated MNBs as a negative control.
These results suggest that this capture procedure is a useful tool for recovery and enrichment of infectious rotavirus. Moreover, when combined with a suitable virus assay this capture procedure can increase the sensitivity of rotavirus detection. Therefore, this capture method is a valuable tool for generating vaccines as well as for developing sensitive detection systems for viruses.
RNA methyltransferases (MTases) catalyse the transfer of a methyl group to their RNA substrates using most-often S-adenosyl-L-methionine (SAM) as cofactor. Only few RNA-bound MTases structures are currently available due to the difficulties in crystallising RNA:protein complexes. The lack of complex structures results in poorly understood RNA recognition patterns and methylation reaction mechanisms. On the contrary, many cofactor-bound MTase structures are available, resulting in well-understood protein:cofactor recognition, that can guide the design of bisubstrate analogues that mimic the state at which both the substrate and the cofactor is bound. Such bisubstrate analogues were recently synthesized for proteins monomethylating the N6-atom of adenine (m6A). These proteins include, amongst others, RlmJ in E. coli and METLL3:METT14 and METTL16 in human. As a proof-of-concept, we here test the ability of the bisubstrate analogues to mimic the substrate:cofactor bound state during catalysis by studying their binding to RlmJ using differential scanning fluorimetry, isothermal titration calorimetry and X-ray crystallography. We find that the methylated adenine base binds in the correct pocket, and thus these analogues could potentially be used broadly to study the RNA recognition and catalytic mechanism of m6A MTases. Two bisubstrate analogues bind RlmJ with micro-molar affinity, and could serve as starting scaffolds for inhibitor design against m6A RNA MTases. The same analogues cause changes in the melting temperature of the m1A RNA MTase, TrmK, indicating non-selective protein:compound complex formation. Thus, optimization of these molecular scaffolds for m6A RNA MTase inhibition should aim to increase selectivity, as well as affinity.
RNA helicases, highly conserved enzymes, are currently believed to be not only involved in RNA modulation, but also in other biological processes. We recently reported that RNA helicase DEAD-box protein 5 (DDX5) is required for maintaining the homeostasis of vascular smooth muscle cells (SMCs). However, the expression and function of RNA helicase in vascular physiology and disease is unknown.
To investigate the role of RNA helicase in vascular diseases.
We showed here that DDX5 was the most abundant DEAD-box protein expressed in human and rodent artery, which mainly located in SMCs. It was demonstrated that DDX5 levels were reduced in cytokine-stimulated SMCs and vascular lesions. DDX5 knocking down or deficiency increased SMC proliferation and migration, while overexpression of DDX5 prevented aberrant proliferation and migration of SMCs. Mechanistic studies revealed transcription factor GATA binding protein 6 (GATA-6) as a novel downstream target of DDX5, which directly interacted with GATA-6 and protected it from mouse double minute 2-mediated degradation. Our ChIP assay identified a previously unreported binding of p27Kip1 promoter to GATA-6. DDX5 increased the recruitment of GATA-6 to p27Kip1 promoter, which enhanced p27Kip1 expression and maintained SMC quiescence. Finally, we showed exacerbated neointima formation in DDX5 SMC-deficient mice after femoral artery injury, whereas overexpression of DDX5 potently inhibited vascular remodeling in balloon injured rat carotid artery.
These findings provide the first evidence for a role of RNA helicase DDX5 in the protection against SMC proliferation, migration, and neointimal hyperplasia. Our data extend the fundamental role of RNA helicase beyond RNA modulation, which provides the basic information for new therapeutic strategies for vascular diseases.
A growing body of evidence implicates the mycobacterial capsule - the outermost layer of the mycobacterial cell envelope - in modulation of the host immune response and virulence of mycobacteria. Mycobacteria synthesize the dominant capsule component, α(1→4)-linked glucan, via three interconnected, and potentially redundant metabolic pathways. Here, we report the crystal structure of the Mycobacterium smegmatis TreS-Pep2 complex, containing trehalose synthase (TreS) and maltokinase (Pep2), which convert trehalose to maltose-1-phosphate as part of the TreS-Pep2-GlgE pathway. The structure, at 3.6 Å resolution, revealed that a diamond-shaped TreS tetramer forms the core of the complex, and that pairs of Pep2 monomers bind to opposite apices of the tetramer in a 4+4 configuration. However, for the M. smegmatis orthologues, results from isothermal titration calorimetry and analytical ultracentrifugation experiments indicated that the prevalent stoichiometry in solution is 4 TreS + 2 Pep2 protomers. The observed discrepancy between the crystallised complex and the behaviour in the solution state may be explained by the relatively weak affinity of Pep2 for TreS (Kd  3.5 μM at mildly acidic pH) and crystal packing favouring the 4+4 complex. Proximity of the ATP binding site in Pep2 to the complex interface provides a rational basis for rate enhancement of Pep2 upon binding to TreS, but the complex structure appears to rule out substrate channeling between the active sites of TreS and Pep2. Our findings provide a structural model for the trehalose synthase-maltokinase complex in M. smegmatis that offers critical insights into capsule assembly.
N6-methyladenosine (m6A) is an abundant mRNA modification that affects multiple biological processes, including those involved in the cell stress response and viral infection. YTH domain family 2 (YTHDF2) is an m6A-binding protein that affects the localization and stability of targeted mRNA. RNA-binding proteins (RBPs) can regulate the stability of inflammatory gene mRNA transcripts, thus participating in the regulation of inflammatory processes. As an RBP, the role of YTHDF2 in the LPS-induced inflammatory reaction has not been reported. To elucidate the function of YTHDF2 in the inflammatory response of macrophages, we first detected the expression level of YTHDF2 in RAW 264.7 cells, and found that it was upregulated after LPS stimulation. YTHDF2 knockdown significantly increased the LPS-induced IL-6, TNF-α, IL-1β, and IL-12 expression and the phosphorylation of p65, p38, and ERK1/2 in NF-κB and MAPK signaling. Moreover, the upregulated expression of TNF-α and IL-6 in cells with silenced YTHDF2 expression was downregulated by the NF-κB, p38, and ERK inhibitors. YTHDF2 depletion increased the expression and stability of MAP2K4 and MAP4K4 mRNAs. All of these results suggest that YTHDF2 knockdown increases mRNA expression levels of MAP2K4 and MAP4K4 via stabilizing the mRNA transcripts, which activate MAPK and NF-κB signaling pathways, which promote the expression of proinflammatory cytokines and aggravate the inflammatory response in LPS-stimulated RAW 264.7 cells.
Prokaryotes have an essential gene-gyrase-that catalyzes negative supercoiling of plasmid and chromosomal DNA. Negative supercoils influence DNA replication, transcription, homologous recombination, site-specific recombination, genetic transposition and sister chromosome segregation. Although E. coli and Salmonella Typhimurium are close relatives with a conserved set of essential genes, E. coli DNA has a supercoil density 15% higher than Salmonella, and E. coli cannot grow at the supercoil density maintained by wild type (WT) Salmonella. E. coli is addicted to high supercoiling levels for efficient chromosomal folding. In vitro experiments were performed with four gyrase isoforms of the tetrameric enzyme (GyrA₂:GyrB₂). E. coli gyrase was more processive and faster than the Salmonella enzyme, but Salmonella strains with chromosomal swaps of E. coli GyrA lost 40% of the chromosomal supercoil density. Reciprocal experiments in E. coli showed chromosomal dysfunction for strains harboring Salmonella GyrA. One GyrA segment responsible for dis-regulation was uncovered by constructing and testing GyrA chimeras in vivo. The six pinwheel elements and the C-terminal 35⁻38 acidic residues of GyrA controlled WT chromosome-wide supercoiling density in both species. A model of enzyme processivity modulated by competition between DNA and the GyrA acidic tail for access to β-pinwheel elements is presented.
The effects of RNA on in-cell NMR spectroscopy and ribosomes on the kinetic activity of several metabolic enzymes are reviewed. Quinary interactions between labelled target proteins and RNA broaden in-cell NMR spectra yielding apparent megadalton molecular weights in-cell. The in-cell spectra can be resolved by using cross relaxation-induced polarization transfer (CRINEPT), heteronuclear multiple quantum coherence (HMQC), transverse relaxation-optimized, NMR spectroscopy (TROSY). The effect is reproduced in vitro by using reconstituted total cellular RNA and purified ribosome preparations. Furthermore, ribosomal binding antibiotics alter protein quinary structure through protein-ribosome and protein-mRNA-ribosome interactions. The quinary interactions of Adenylate kinase, Thymidylate synthase and Dihydrofolate reductase alter kinetic properties of the enzymes. The results demonstrate that ribosomes may specifically contribute to the regulation of biological activity.
Virus-inhibitory protein, endoplasmic reticulum-associated, interferon-inducible (viperin) is a radical SAM enzyme that plays a multifaceted role in the cellular antiviral response. Viperin has recently been shown to catalyze the SAM-dependent formation of 3'-deoxy-3',4'-didehydro-CTP (ddhCTP), which inhibits some viral RNA polymerases. Viperin is also implicated in regulating Lys-63-linked polyubiquitination of interleukin-1 receptor-associated kinase-1 (IRAK1) by the E3 ubiquitin ligase tumor necrosis factor receptor-associated factor 6 (TRAF6) as part of the Toll-like receptor-7 and -9 (TLR7/9) innate immune signaling pathways. In these pathways, the poly-ubiquitination of IRAK1 by TRAF6 is necessary to activate IRAK1, which then phosphorylates downstream targets and ultimately leads to the production of type I interferons. That viperin is a component of these pathways suggested that its enzymatic activity might be regulated by interactions with partner proteins. To test this idea, we have reconstituted the interactions between viperin, IRAK1, and TRAF6 by transiently expressing these enzymes in HEK 293T cells. We show that IRAK1 and TRAF6 increase viperin activity ∼10-fold to efficiently catalyze the radical-mediated dehydration of CTP to ddhCTP. Furthermore, we found that TRAF6-mediated ubiquitination of IRAK1 requires the association of viperin with both IRAK1 and TRAF6. Ubiquitination appears to depend on structural changes in viperin induced by SAM binding, but, significantly, does not require catalytically active viperin. We conclude that the synergistic activation of viperin and IRAK1 provides a mechanism that couples innate immune signaling with the production of the antiviral nucleotide ddhCTP.
N6-methyladenosine (m6A) mRNA modifications play critical roles in various biological processes. However, no study addresses the role of m6A in macroautophagy/autophagy. Here, we show that m6A modifications are increased in H/R-treated cardiomyocytes and ischemia/reperfusion (I/R)-treated mice heart. We found that METTL3 (methyltransferase like 3) is the primary factor involved in aberrant m6A modification. Silencing METTL3 enhances autophagic flux and inhibits apoptosis in H/R-treated cardiomyocytes. However, overexpression of METTL3 or inhibition of the RNA demethylase ALKBH5 has an opposite effect, suggesting that METTL3 is a negative regulator of autophagy. Mechanistically, METTL3 methylates TFEB, a master regulator of lysosomal biogenesis and autophagy genes, at two m6A residues in the 3'-UTR, which promotes the association of the RNA-binding protein HNRNPD with TFEB pre-mRNA and subsequently decreases the expression levels of TFEB. Further experiments show that autophagic flux enhanced by METTL3 deficiency is TFEB dependent. In turn, TFEB regulates the expression levels of METTL3 and ALKBH5 in opposite directions: it induces ALKBH5 and inhibits METTL3. TFEB binds to the ALKBH5 promoter and activates its transcription. In contrast, inhibition of METTL3 by TFEB does not involve transcriptional repression but rather downregulation of mRNA stability, thereby establishing a negative feedback loop. Together, our work uncovers a critical link between METTL3-ALKBH5 and autophagy, providing insight into the functional importance of the reversible mRNA m6A methylation and its modulators in ischemic heart disease. Abbreviations: ACTB, actin beta; ALKBH5, alkB homolog 5, RNA demethylase; ANXA5, annexin A5; ATG, autophagy-related; BafA, bafilomycin A1; CASP3, caspase 3; ELAVL1, ELAV like RNA binding protein 1; FTO, FTO, alpha-ketoglutarate dependent dioxygenase; GFP, green fluorescent protein; GST, glutathione S-transferase; HNRNPD, heterogeneous nuclear ribonucleoprotein D; H/R, hypoxia/reoxygenation; I/R, ischemia/reperfusion; LAD, left anterior descending; m6A, N6-methyladenosine; MEFs, mouse embryo fibroblasts; Mer, mutated estrogen receptor domains; METTL3, methyltransferase like 3; METTL14, methyltransferase like 14; mRFP, monomeric red fluorescent protein; MTORC1, mechanistic target of rapamycin kinase complex 1; NMVCs, neonatal mouse ventricular cardiomyocytes; PCNA, proliferating cell nuclear antigen; PE, phosphatidylethanolamine; PI, propidium iodide; PTMs, post-translational modifications; PVDF, polyvinylidenedifluoride; RIP, RNA-immunoprecipitation; siRNA, small interfering RNA; SQSTM1, sequestosome 1; TFEB, transcription factor EB; TUBA: tublin alpha; WTAP, WT1 associated protein; YTHDF, YTH N6-methyladenosine RNA binding protein.
Vascular calcification is highly prevalent in end-stage renal diseases and is predictive of cardiovascular events and mortality. Poly(ADP-ribose) polymerase 1 (PARP1) inhibition or deletion is vasoprotective in several disease models. Here we show that PARP activity is increased in radial artery samples from patients with chronic renal failure, in arteries from uraemic rats, and in calcified vascular smooth muscle cells (VSMCs) in vitro. PARP1 deficiency blocks, whereas PARP1 overexpression exacerbates, the transdifferentiation of VSMCs from a contractile to an osteogenic phenotype, the expression of mineralization-regulating proteins, and calcium deposition. PARP1 promotes Runx2 expression, and Runx2 deficiency offsets the pro-calcifying effects of PARP1. Activated PARP1 suppresses miRNA-204 expression via the IL-6/STAT3 pathway and thus relieves the repression of its target, Runx2, resulting in increased Runx2 protein. Together, these results suggest that PARP1 counteracts vascular calcification and that therapeutic agents that influence PARP1 activity may be of benefit to treat vascular calcification.
Bone marrow stem cells (BMSCs) are multipotent stem cells that can regenerate mesenchymal tissues, such as adipose tissue, bone, and muscle. Recent studies have shown that N6-methyladenosine (m6A) methylation, one of the most prevalent epigenetic modifications, is involved in the development process. However, whether it plays roles in BMSC differentiation is still elusive. Here, we found that the deletion of m6A "writer" protein methyltransferase-like (METTL)3 in porcine BMSCs (pBMSCs) could promote adipogenesis and janus kinase (JAK)1 protein expression via an m6A-dependent way. Knockdown of METTL3 decreased mRNA m6A levels of JAK1, leading to enhanced YTH m6A RNA binding protein 2 (YTHDF2)-dependent JAK1 mRNA stability. We further demonstrated that JAK1 activated signal transducer and activator of transcription (STAT) 5 through regulation of its phosphorylation to bind to the promoter of CCAAT/enhancer binding protein (C/EBP) β, which could ultimately lead to a modulated adipogenic process. Collectively, our results reveal an orchestrated network linking the m6A methylation and JAK1/STAT5/C/EBPβ pathway in pBMSCs adipogenic differentiation. Our findings provide novel insights into the underlying molecular mechanisms of m6A modification in the regulation of BMSCs differentiating into adipocytes, which may pave a way to develop more effective therapeutic strategies in stem cell regenerative medicine and the treatment of obesity.-Yao, Y., Bi, Z., Wu, R., Zhao, Y., Liu, Y., Liu, Q., Wang, Y., Wang, X. METTL3 inhibits BMSC adipogenic differentiation by targeting the JAK1/STAT5/C/EBPβ pathway via an m6A-YTHDF2-dependent manner.
The hepatitis C virus (HCV) nonstructural protein 5A (NS5A) plays a key role in viral replication and virion assembly, and regulation of the assembly process critically depends on phosphorylation of both serine and threonine residues in NS5A. We previously identified SRC proto-oncogene, non-receptor tyrosine kinase (c-Src) as an essential host component of the HCV replication complex consisting of NS5A, the RNA-dependent RNA polymerase NS5B, and c-Src. Pull-down assays revealed an interaction between NS5A and the Src-homology 2 (SH2) domain of c-Src; however, the precise binding mode remains undefined. In this study, using a variety of biochemical and biophysical techniques, along with molecular dynamics simulations, we demonstrate that the interaction between NS5A and the c-Src SH2 domain strictly depends on an intact, phosphotyrosine binding-competent SH2 domain and on tyrosine phosphorylation within NS5A. Detailed analysis of c-Src SH2 domain binding to a panel of phosphorylation-deficient NS5A variants revealed that phosphorylation of Y93 located within domain 1 of NS5A, but not of any other tyrosine residue, is crucial for complex formation. In line with these findings, effective replication of subgenomic HCV replicons as well as production of infectious virus particles in mammalian cell culture models were clearly dependent on the presence of tyrosine at position 93 of NS5A. These findings indicate that phosphorylated Y93 in NS5A plays an important role during viral replication by facilitating NS5A's interaction with the SH2 domain of c-Src.
Our study aims to explore whether microRNA-940 could participate in early abortion by inhibiting placenta implantation and its underlying mechanism.
Expressions of microRNA-940 and ZNF672 in the placental villi of 6 early abortion pregnancies and 6 normal pregnancies were detected by quantitative Real Time-Polymerase Chain Reaction (qRT-PCR). Expressions of microRNA-940 and ZNF672 in trophoblast cells (BeWo, JEG3, Wish and HTR-8) were also detected. Cell counting kit-8 (CCK-8) assay was performed to detect the effect of microRNA-940 on the proliferation of trophoblast cells after being transfected with microRNA-940 inhibitor or mimics, respectively. Dual-Luciferase reporter gene assay and RNA binding protein immunoprecipitation (RIP) assay were conducted to demonstrate the binding condition of microRNA-940 to ZNF672.
MicroRNA-940 was highly expressed in placental villi of early abortion pregnancies, whereas ZNF672 was lowly expressed. HTR-8 cells expressed the highest level and BeWo cells expressed the lowest level of microRNA-940. After transfection of microRNA-940 inhibitor in HTR-8 cells, the proliferative capacity was remarkably increased. On the contrary, the transfection of microRNA-940 mimic downregulated the proliferation of BeWo cells. Dual-Luciferase reporter gene assay demonstrated that microRNA-940 targets ZNF672. RIP results further indicated that microRNA-940 binds to ZNF672.
MicroRNA-940 is highly expressed in the placental villi of early abortion pregnancies and promotes the occurrence of early abortion by inhibiting the proliferation of trophoblast cells by targeting ZNF672.
Extracellular cold-inducible RNA-binding protein (CIRP) exaggerates inflammation and tissue injury in sepsis. Neutrophil extracellular traps (NETs) are released by activated neutrophils during sepsis. NETs contribute to pathogen clearance, but excessive NET formation (NETosis) causes inflammation and tissue damage. Peptidylarginine deiminase 4 (PAD4) is associated with NETosis by increasing histone citrullination and chromatin decondensation. We hypothesized that CIRP induces NETosis in the lungs during sepsis via upregulating PAD4 expression. Sepsis was induced in C57BL/6 wild-type (WT) and CIRP-/- mice by cecal ligation and puncture (CLP). After 20 h of CLP induction, NETs in the lungs of WT and CIRP-/- mice were quantified by flow cytometry by staining the single cell suspensions with MPO and CitH3 Abs. PAD4 expression in the lungs of WT and CIRP-/- mice after sepsis was assessed by Western blotting. In vitro effects of recombinant mouse (rm) CIRP for NETosis and PAD4 expression in the bone marrow-derived neutrophils (BMDN) were assessed by flow cytometry and Western blotting, respectively. After 20 h of CLP, NETosis in the lungs was significantly decreased in CIRP-/- mice compared to WT mice, which also correlated with the decreased PAD4 expression. Intratracheal administration of rmCIRP into WT mice significantly increased NETosis and PAD4 expression in the lungs compared to vehicle-injected mice. In vitro culture of BMDN with rmCIRP significantly increased NETosis and PAD4 expression compared to PBS-treated control. Fluorescence microscopy revealed typical web-like structures consistent with NETs in rmCIRP-treated BMDN. Thus, CIRP serves as a novel inducer of NETosis via PAD4 during sepsis.
Ribosome biogenesis has been implicated in resistance exercise training (RET)-induced skeletal muscle hypertrophy. However, it is unclear how increasing bouts of RET affects ribosome content and biogenesis. This was investigated in the present study using simulated RET where rat skeletal muscle is subjected to increasing bouts of electrical stimulations. Sprague-Dawley rats were randomly assigned to the following seven groups: sedentary for 5 days (SED) or 6 weeks (SED_6w), resistance-exercise trained with 1 bout (1B), 2 bouts (2B), 3 bouts (3B), 6 bouts (6B), and 18 bouts (18B). RET was simulated on the right gastrocnemius muscle by transcutaneous electric stimulation under isoflurane anesthesia, and RET bout was given 3 times a week. Rats in 1B, 2B and 3B groups showed increased 45S precursor (pre-) rRNA and 18S+28S rRNA content per muscle weight and elevated mRNA levels of c-myc and upstream binding factor (UBF). Increases in phosphorylated UBF and total cyclin D1 protein level were observed 48 h after RET; the former increased as a function of RET duration. In 3B, 6B and 18B groups, the 18S+28S rRNA content per muscle weight was kept unchanged, and 45S pre-rRNA, cyclin D1 and phosphorylated UBF levels in 18B were lower than those in 3B. These results suggest that RET activates ribosome biogenesis and increases ribosome content through modulation of UBF and cyclin D1 activity at its early phase. Additional bouts of RET may not lead to further increase in ribosome content per muscle weight through possibly the attenuation of transcription process.
Chronic inflammation is an important etiologic mechanism for muscle atrophy. Oat-derived phytochemical avenanthramides (AVAs) have been shown to suppress inflammatory responses in human clinical studies and in several cell lines in vitro, but their role in skeletal muscle is unclear. The aim of this study was to investigate whether AVA treatment can prevent tumor necrosis factor (TNF)-α-induced muscle fiber atrophy in C2C12 cells.
We treated 70% confluent cells for 24 h with AVA. Then, TNF-α was added to cell-cultured medium. Subsequently, cells were harvested at different time points. The cells were examined using various biochemical techniques for measuring protein, messenger RNA levels, nuclear binding activity, and viability. Fluorescence microscope was used for analysis of the myotube morphology.
Cells treated with TNF-α significantly increased nuclear factor κB activation, indicated by a marked decrease of IκB (p &lt; 0.05) and a 6.6-fold increase in p65-DNA binding (p &lt; 0.01); however, 30 μmol of AVA-A, -B, and -C treatment reduced the binding by 33%, 18%, and 19% (p &lt; 0.01), respectively, compared with cells treated with TNF-α without AVA. The interleukin-6 level increased by 2.5 fold (p &lt; 0.01) with TNF-α, but decreased by 24%, 32%, and 28% (p &lt; 0.01), respectively, with AVA-A, -B, and -C. The interleukin-1β level also showed a 47% increase with TNF-α (p &lt; 0.01), whereas this increment was abolished in all AVA-treated cells. Reactive oxygen species production was 1.3-fold higher in the TNF-α-treated group (p &lt; 0.01) but not in the TNF-α + AVAs groups. Messenger RNA levels of muscle-specific E3 ubiquitin ligase atrogin-1 increased 23% in TNF-α vs. control (p &lt; 0.05) but was decreased by 46%, 34%, and 53% (p &lt; 0.01), respectively, with treatment of AVA-A, -B, and -C. Moreover, TNF-α treatment increased the muscle RING finger 1 messenger RNA level by 76% (p &lt; 0.01); this change was abolished by AVAs. Cells treated with TNF-α demonstrated a reduced proliferation compared with control cells (p &lt; 0.01), but this effect was not seen in TNF-α + AVAs cells. The diameter of the C2C12 myotube decreased by 28% (p &lt; 0.01) with TNF-α, whereas it showed no change when AVAs were included in the cell media.
These results indicated that AVAs can reduce proinflammatory cytokine and reactive oxygen species production and ameliorate TNF-α-induced myotube atrophy in muscle cells.
Infection and inflammation are able to induce diet-independent Na+-accumulation without commensurate water retention in afflicted tissues, which favors the pro-inflammatory activation of mouse macrophages and augments their antibacterial and antiparasitic activity. While Na+-boosted host defense against the protozoan parasite Leishmania major is mediated by increased expression of the leishmanicidal NOS2 (nitric oxide synthase 2, inducible), the molecular mechanisms underpinning this enhanced antibacterial defense of mouse macrophages with high Na+ (HS) exposure are unknown. Here, we provide evidence that HS-increased antibacterial activity against E. coli was neither dependent on NOS2 nor on the phagocyte oxidase. In contrast, HS-augmented antibacterial defense hinged on HIF1A (hypoxia inducible factor 1, alpha subunit)-dependent increased autophagy, and NFAT5 (nuclear factor of activated T cells 5)-dependent targeting of intracellular E. coli to acidic autolysosomal compartments. Overall, these findings suggest that the autolysosomal compartment is a novel target of Na+-modulated cell autonomous innate immunity.
ACT: actins; AKT: AKT serine/threonine kinase 1; ATG2A: autophagy related 2A; ATG4C: autophagy related 4C, cysteine peptidase; ATG7: autophagy related 7; ATG12: autophagy related 12; BECN1: beclin 1; BMDM: bone marrow-derived macrophages; BNIP3: BCL2/adenovirus E1B interacting protein 3; CFU: colony forming units; CM-H2DCFDA: 5-(and-6)-chloromethyl-2',7'-dichlorodihydrofluorescein diacetate, acetyl ester; CTSB: cathepsin B; CYBB: cytochrome b-245 beta chain; DAPI: 4,6-diamidino-2-phenylindole; DMOG: dimethyloxallyl glycine; DPI: diphenyleneiodonium chloride; E. coli: Escherichia coli; FDR: false discovery rate; GFP: green fluorescent protein; GSEA: gene set enrichment analysis; GO: gene ontology; HIF1A: hypoxia inducible factor 1, alpha subunit; HUGO: human genome organization; HS: high salt (+ 40 mM of NaCl to standard cell culture conditions); HSP90: heat shock 90 kDa proteins; LDH: lactate dehydrogenase; LPS: lipopolysaccharide; Lyz2/LysM: lysozyme 2; NFAT5/TonEBP: nuclear factor of activated T cells 5; MΦ: macrophages; MAP1LC3/LC3: microtubule associated protein 1 light chain 3; MFI: mean fluorescence intensity; MIC: minimum inhibitory concentration; MOI: multiplicity of infection; MTOR: mechanistic target of rapamycin kinase; NaCl: sodium chloride; NES: normalized enrichment score; n.s.: not significant; NO: nitric oxide; NOS2/iNOS: nitric oxide synthase 2, inducible; NS: normal salt; PCR: polymerase chain reaction; PGK1: phosphoglycerate kinase 1; PHOX: phagocyte oxidase; RFP: red fluorescent protein; RNA: ribonucleic acid; ROS: reactive oxygen species; sCFP3A: super cyan fluorescent protein 3A; SBFI: sodium-binding benzofuran isophthalate; SLC2A1/GLUT1: solute carrier family 2 (facilitated glucose transporter), member 1; SQSTM1/p62: sequestosome 1; ULK1: unc-51 like kinase 1; v-ATPase: vacuolar-type H+-ATPase; WT: wild type.
The metastasis suppressor, N-myc downstream regulated gene-1 (NDRG1), is a stress response protein that is involved in the inhibition of multiple oncogenic signaling pathways. Initial studies have linked NDRG1 and the endoplasmic reticulum (ER) stress response. Considering this, we extensively examined the mechanism by which NDRG1 regulates the ER stress response in pancreatic and colon cancer cells. We also examined the anti-cancer agent, di-2-pyridylketone 4,4-dimethyl-3-thiosemicarbazone (Dp44mT), which induces NDRG1 expression and causes ER stress. The expression of NDRG1 was demonstrated to regulate the three main arms of the ER stress response by: (1) increasing the expression of three major ER chaperones, binding immunoglobulin protein (BiP), calreticulin, and calnexin; (2) suppressing the protein kinase, RNA-activated (PKR)-like ER kinase (PERK); (3) inhibiting the inositol-requiring kinase 1α (IRE1α) arm; and (4) increasing the cleavage of activating transcription factor 6 (ATF6). An important finding was that NDRG1 enhances the anti-proliferative and anti-migratory activity of Dp44mT. This increased efficacy could be related to the following effects in the presence of Dp44mT and NDRG1, namely: markedly increased activation of the PERK target, eukaryotic translation initiation factor 2α (eIF2α); the maintenance of activating transcription factor 4 (ATF4) expression; high cytosolic Ca+2 that increases the sensitivity of cells to apoptosis via activation of the calmodulin-dependent kinase II (CaMKII) signaling cascade; and increased pro-apoptotic C/EBP-homologous protein (CHOP) expression. Collectively, this investigation dissects the molecular mechanisms through which NDRG1 manipulates the ER stress response and its ability to potentiate the activity of the potent anti-cancer agent, Dp44mT.
To determine aberrant circular RNA (circRNA) expression profiles in cumulus cells from polycystic ovarian syndrome (PCOS) patients and identify their potential biological functions.
Circular RNAs microarray analysis of human tissue.
University hospital.
A total of 40 women, including 20 PCOS patients and 20 non-PCOS patients.
None.
A circRNA microarray containing probes that interrogate 21,442 human circRNAs to investigate differentially expressed circRNAs in cumulus cells, with potential target genes of significantly changed circRNAs and biological functions measured by microRNA support vector regression (mirSVR) and gene ontology (GO) analysis, with Kyoto Encyclopedia of Genes and Genomes (KEGG) pathway analysis.
A total of 1,032 circRNAs were identified that were differentially expressed in PCOS cumulus cells, including 311 circRNAs increase and 721 circRNAs decrease (fold change ≥2). Four aberrantly expressed circRNAs reached a statistically significant result after Bonferroni correction (with Bonferroni correction, only circRNAs for which P &lt; .05/21,442 = 2.3 × 10-6 were considered statistically significant). Further analysis showed that aberrantly expressed circRNAs harbored microRNA binding sites, and some microRNAs were associated with PCOS. The GO and KEGG biological pathway analysis indicated that the genes with protein binding, mitotic nuclear envelope disassembly and metabolic pathways were statistically significantly enriched.
Our data suggest that the aberrantly expressed circRNAs and their targeted genes might be associated with PCOS, providing new clues to find key diagnostic and therapeutic molecular biomarkers for PCOS patients.
Recent evidence has suggested that cadmium (Cd) ions-induced neurotoxicity is associated with increased oxidative stress and mitochondrial-dependent and endoplasmic reticulum (ER) stress-induced apoptosis. This study aimed to investigate if rutin hydrate (RH), a well-reported neuroprotective and an antioxidant flavonoid, can ameliorate cadmium chloride (CdCl2)-induced neurotoxicity by inhibiting the resultant ER stress. Rats were divided into 4 groups (n = 16/group) of control, control + RH (100 mg/kg), CdCl2 (5 mg/kg), and CdCl2 + RH. All treatments were administered orally for 30 days, on daily basis. Brain homogenates from CdCl2-treated rats showed increased oxidative stress and induced activation of ER stress characterized by increasing mRNA and protein levels of GRP78, ATF-6, CHOP and Xbp-1 and protein levels of p-elF2α, p-JNK1/2 and cleaved caspase-12. Also, CdCl2 significantly reduced Bcl-2, enhanced Bax translocation to the mitochondrial membrane, increased cytoplasmic levels of cytochrome-C and caspase-3, and reduced mitochondrial membrane potential (Δψm) (increased Vmax and reduced time to Vmax). In contrast, RH significantly enhanced levels GSH and activities of SOD, GSH-Px, decreased levels of MDA and inhibited mitochondrial permeability transition pore (mtPTP) in the brains of both control and CdCl2-treated rats. Interestingly, in brain homogenates of CdCl2-treated rats only, RH reduced all markers of ER stress, increased Bcl-2, reduced mitochondrial Bax translocation and improved mitochondrial coupling. It also reduced cytosolic levels of cytochrome-C, cleaved caspase-3, and cleaved caspase-12. Overall, these findings support the efficiency of RH to inhibit ER stress in brains CdCl2-treated rats which is added to its existing mechanisms of neuroprotection. Abbreviations: ATF-6: activating transcription factor-6; Bax: Bcl-associated x; BBB: blood-brain barrier; Bcl-2: B-cell lymphoma 2; BiP: immunoglobulin heavy-chain-binding protein; [Ca2+]i: intracellular free Ca2+ concentration; Cd: cadmium; CdCl2: cadmium chloride; CHOP: CCAAT/enhancer-binding protein-homologous protein; CMC: carboxymethyl cellulose; Δψm: mitochondrial membrane potential; elF2α: phospho-eukaryotic translation initiation factor 2-alpha; ER: endoplasmic reticulum; ERAD: ER-associated protein degradation; ERK1/2: extracellular signal-regulated kinases 1 and 2; GADD 153: growth arrest and DNA damage-inducible protein 153; GRP78, 78 kDa glucose-regulated protein; GSH: reduced glutathione; GSH: reduced glutathione; GSH-Px: glutathione peroxidase; GSSG: glutathione disulfide (oxidized glutathione); IRE-1: inositol-requiring enzyme-1; JNK: c-Jun N-terminal kinase; MAPK: mitogen-activated protein kinase; MDA: malondialdehyde; mTOR: Akt/mammalian target of rapamycin; mtPTP: mitochondrial permeability transition pore; ONOO-: peroxynitrite; PCR: polymerase chain reaction; PERK: protein kinase RNA-like ER kinase; p-JNK: phospho-JNK; qPCR: quantitative PCR; RCR: respiratory control ratio; RH: rutin hydrate; RHoGDI: Rho-GDP-dissociation inhibitor; ROS: reactive oxygen species; SOD: superoxide dismutase; UPR: unfolded protein response; VDAC: voltage-dependent anion channel; Vmax: maximal rate of pore opening; Xbp-1: X-box binding protein 1.
To determine the correlation between QKI and pancreatic cancer tissues, the QKI expression of pancreatic cancer cells and fibroblasts in the tumor-surrounding microenvironment were detected. Then, QKI overexpression and interference with QKI short hairpin RNA in LX-2 (a fibroblast cell line) were established in vitro. Meanwhile, to observe the cell proliferation, invasion, migration, and other changes, QKI, and related epithelial-mesenchymal transition (EMT) molecules were detected by a polymerase chain reaction and Western blot analysis. In addition, an in vivo tumorigenicity test in node mice was performed to confirm whether QKI expression can promote the proliferation, invasion, and metastasis of pancreatic cancer ductal epithelial cells. Finally, the autophagy levels of fibroblasts with QKI overexpression were observed by electron microscopy to further explore the QKI pathogenic mechanism. It was found that cell proliferation, invasion, migration, and EMT-related markers were increased in QKI-overexpressed fibroblasts LX-2. Furthermore, in vivo, liver and peritoneal metastasis decreased overall survival rate and increasing autophagy levels in QKI-overexpressing nude mice were observed. Meanwhile, knock down QKI with small interfering RNA can reverse all the above effects. QKI can promote the proliferation, metastasis, and invasion of pancreatic cancer through activating fibroblasts surrounding pancreatic cancer and accelerating EMT and increasing the autophagy in pancreatic cancer. QKI may become a potential target for the treatment of pancreatic cancer.
Increased visceral adipose tissue and dysbiosis in the overweight and obese promote chronic inflammation. The aim of this study was to compare the effects of moderate-intensity continuous training (MICT) and high-intensity interval training (HIIT) on the gut-adipose tissue cross-talk in obese Zucker rats.
Obese male Zucker rats (n = 36) were divided in three groups: MICT (12m.min-1 for 51min), HIIT (6 sets at 18 m.min-1 for 4min followed by 3min at 10m.min-1) and controls (CONT; no exercise). The animals ran on a treadmill 5 days/week for 10 weeks. Body composition, glycaemic control, lipid profile, inflammation, lipolysis signalling in subcutaneous and visceral adipose tissue, intestinal permeability (tight junctions and plasma lipopolysaccharide binding protein; LBP), and gut microbiota composition were assessed in the three groups.
After 10 weeks of exercise, total and epididymal fat mass decreased only in the HIIT group. The α/β adrenergic receptor RNA ratio in subcutaneous adipose tissue increased only in the HIIT group. The expression level of phosphorylated hormone-sensitive lipase was not modified by training. Both HIIT and MICT decreased inflammation (plasma myeloperoxidase and keratinocyte-derived chemokine secretion in adipose tissue) and improved glucose metabolism. Zonula occludens-1 and occludin were upregulated in the HIIT group. Plasma LBP was similarly reduced in both training groups. HIIT and MICT did not affect gut microbiota composition.
In obese Zucker rats, HIIT and MICT improved inflammation and glucose metabolism. In contrast, only HIIT decreased total and visceral fat mass. These adaptations were not associated with modifications in gut microbiota composition.
Autophagy, a lysosomal self-degradative process of cellular components, is essential for cellular homeostasis to response cellular stress and is tightly controlled by autophagy-related genes (ATGs). Autophagy-related gene 6 (ATG6, also known as Beclin-1 in human) is an essential factor regulating autophagy and apoptosis. RNA binding proteins (RBPs) regulate gene expression at the post-transcriptional level and their differential expression is linked to the pathogenesis of several human diseases. Here, we demonstrate the role of heterogeneous nuclear ribonucleoprotein A1 (hnRNPA1) as a novel factor regulating ATG6 expression. hnRNPA1 associates with the 3' untranslated region (3'UTR) of ATG6 mRNA and promotes its expression without significant changes at the mRNA level. Knockdown of hnRNPA1 decreases ATG6 expression, which is enhanced by the overexpression of hnRNPA1. Also, we show augmented expression of both hnRNPA1 and ATG6 in the colorectal cancer (CRC) tissues obtained from patients and demonstrate a positive correlation of their expression in CRC tissues. Our results suggest the potential role of hnRNPA1-mediated ATG6 regulation in the pathogenesis of CRC.
CCAAT/enhancer binding protein epsilon (CEBPE) is an essential transcription factor for granulocytic differentiation. Mutations of CEBPE occur in individuals with neutrophil-specific granule deficiency (SGD), which is characterized by defects in neutrophil maturation. Cebpe knockout mice also exhibit defects in terminal differentiation of granulocytes, a phenotype reminiscent of SGD. Analysis of DNase-Seq data revealed an open chromatin region 6kb downstream of the transcriptional start site of Cebpe in murine myeloid cells. We identified interaction between this +6 kb region and the core promoter of Cebpe using 4C-seq. To understand the role of this putative enhancer in transcriptional regulation of Cebpe, we targeted it using dCas9-KRAB and observed a significant downregulation of transcript and protein levels of CEBPE in cells expressing guide RNA targeting the +6 kb region. To investigate further the role of this novel enhancer in myelopoiesis, we generated mice with deletion of this region using CRISPR/Cas9 technology. Germline deletion of the +6 kb enhancer resulted in reduced levels of CEBPE and its target genes and caused a severe block in granulocytic differentiation. We also identified binding of CEBPA and CEBPE to the +6 kb enhancer which suggests their role in regulating the expression of Cebpe. In summary, we have identified a novel enhancer crucial for regulating expression of Cebpe, and required for normal granulocytic differentiation.
Colorectal cancer is one of the most prevalent malignancies worldwide. ENKUR is a transient receptor potential canonical-binding protein that acts as a potential regulator or effector of transient receptor potential canonical channels. It also directly interacts with the p85 regulatory subunit of phosphoinositide 3-kinase. However, the role of ENKUR in colorectal cancer remains unclear. In the present study, the expression profiles of ENKUR in the The Cancer Genome Atlas and ONCOMINE databases were analyzed. Significant downregulation of ENKUR was observed in clinical tumor samples of various cancer types, including colorectal cancer. Decreased ENKUR messenger RNA expression and ENKUR protein level were detected in 6 human colorectal cancer cell lines. Silencing of ENKUR in colorectal cancer cells led to enhanced cell proliferation, migration, and invasion, while the opposite effects were achieved in ENKUR-overexpressing cells. Furthermore, ENKUR-underexpressing cells exhibited increased activity of phosphoinositide 3-kinase /Akt signaling pathway. Downregulation of the epithelial marker, E-cadherin, and upregulation of the mesenchymal markers, vimentin and N-cadherin, were detected in ENKUR-underexpressing cells, suggesting the induction of epithelial-mesenchymal transition. In conclusion, the present study demonstrates that ENKUR may be responsible for alterations in the proliferative, migratory, and invasive potential of colorectal cancer cells through possible involvement in the phosphoinositide 3-kinase /Akt signaling pathway.
The elongation of very long chain fatty acids protein 6 (ELOVL6) gene encodes a key enzyme that plays a role in lipogenesis through the catalytic elongation of both saturated and monounsaturated fatty acids. Previous studies have described the high expression of bovine ELOVL6 in adipose tissues. However, transcriptional regulation and the functional role of ELOVL6 in lipid metabolism and adipocyte proliferation remain unexplored. Here, a 1.5 kb fragment of the 5'-untranslated region promoter region of ELOVL6 was amplified from the genomic DNA of Qinchuan cattle and sequenced. The core promoter region was identified through unidirectional 5'-end deletion of the promoter plasmid vector. In silico analysis predicted important transcription factors that were then validated through site-directed mutation and small interfering RNA interference with an electrophoretic mobility shift assay. We found that the binding of KLF6 and PU.1 transcription factors occurred in the region -168/+69. Both perform a vital regulatory function in the transcription of bovine ELOVL6. Overexpression of ELOVL6 significantly upregulated the expression of peroxisome proliferator activated receptor γ (PPARγ), but inhibited the expression of fatty acid-binding protein 4 (FABP4), while silencing of ELOVL6 negatively regulated the messenger RNA expression level of PPARγ, FABP4, ACSL, and FATP1. In addition, ELOVL6 promotes adipocyte proliferation by regulating the cell-cycle genes' expression. Taken together, these findings provide useful information about the transcriptional regulation and functional mechanisms of bovine ELOVL6 in lipid metabolism and adipocyte proliferation in Qinchuan cattle.
Class 1 integrons accumulate antibiotic resistance genes by site-specific recombination at aatI 1 sites. Captured genes are transcribed from a promoter located within the integron, for class 1 integrons the first gene to be transcribed and translated normally encodes an aminoglycoside antibiotic resistance protein (either an acetyl- (AAC) or adenyl- (AAD) transferase). The leader RNA from the Pseudomonas fluorescens class 1 integron contains an aminoglycoside sensing riboswitch RNA that controls expression of the downstream aminoglycoside resistance gene. Here we explore the relationship between integron dependent DNA recombination and potential aminoglycoside sensing riboswitch products of recombination derived from a series of aminoglycoside resistant clinical strains. Sequence analysis of clinical strains identified a series of sequence variants that were associated with class I integron derived aminoglycoside resistant (both aac and aad) recombinants. For the aac recombinants, representative sequences showed up to 6-fold aminoglycoside dependent regulation of reporter gene expression. MicroScale Thermophoresis (MST) confirmed RNA binding. Covariance analysis generated a secondary structure model for the RNA that is an independent verification of previous models that were derived from mutagenesis and chemical probing data, and was similar to that of the P. fluorescens riboswitch RNA. The aminoglycosides were among the first antibiotics to be used clinically and the data suggest that in an aminoglycoside rich environment, functional riboswitch recombinants were selected during integron-mediated recombination to regulate aminoglycoside resistance. The incorporation of a functional aminoglycoside sensing riboswitch by integron recombination confers a selective advantage for the expression of resistance genes of diverse origins.
N6-methyladenosine (m6A) is the most abundant internal modification in RNAs and plays regulatory roles in a variety of biological and physiological processes. Despite its important roles, the molecular mechanism underlying m6A-mediated gene regulation is poorly understood. Here, we show that m6A-containing RNAs are subject to endoribonucleolytic cleavage via YTHDF2 (m6A reader protein), HRSP12 (adaptor protein), and RNase P/MRP (endoribonucleases). We demonstrate that HRSP12 functions as an adaptor to bridge YTHDF2 and RNase P/MRP, eliciting rapid degradation of YTHDF2-bound RNAs. Transcriptome-wide analyses show that m6A RNAs that are preferentially targeted for endoribonucleolytic cleavage have an HRSP12-binding site and a RNase P/MRP-directed cleavage site upstream and downstream of the YTHDF2-binding site, respectively. We also find that a subset of m6A-containing circular RNAs associates with YTHDF2 in an HRSP12-dependent manner and is selectively downregulated by RNase P/MRP. Thus, our data expand the known functions of RNase P/MRP to endoribonucleolytic cleavage of m6A RNAs.
</body></html>